<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/elucidata.ico">
  
  <title>Notebooks - Polly Documentation</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Notebooks";
    var mkdocs_page_input_path = "Scaling compute/Polly Notebooks.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href="../index.html" class="icon icon-home"> Polly Documentation</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="../index.html">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Getting Started</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../Getting Started/Workspaces.html">Workspaces</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Apps</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../Apps/Introduction.html">Introduction</a>
                </li>
                <li class="">
                    
    <a class="" href="../Apps/Host Apps.html">Host Apps</a>
                </li>
                <li class="">
                    
    <span class="caption-text">Metabolomic Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Metabolomic Data/El-MAVEN Phi Relative LC-MS.html">El-MAVEN Phi Relative LC-MS</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Metabolomic Data/MetScape.html">MetScape</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Metabolomic Data/FirstView.html">FirstView</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Metabolomic Data/Untargeted Pipeline.html">Untargeted Pipeline</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Metabolomic Data/Dual Mode Visualization.html">Dual Mode Data Visualization</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Metabolomic Data/QuantFit.html">QuantFit</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Metabolomic Data/El-MAVEN.html">El-MAVEN</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Metabolomic Data/Labeled LC-MSMS Workflow.html">Labeled LC-MS/MS Workfow</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Metabolomic Data/Labeled LC-MS Workflow.html">Labeled LC-MS Workflow</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Metabolomic Data/CompoundDiscoverer QuantFit.html">Compound Discoverer<sup>TM</sup> QuantFit</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Screening Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Screening Data/CRISPR Screening.html">CRISPR Screening</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Screening Data/High Throughput Drug Screening.html">High Throughput Drug Screening</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Sequencing Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Sequencing Data/RNA Seq Workflow.html">RNA Seq Workflow</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Proteomic Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Proteomic Data/Proteomics Workflow.html">Proteomics Workflow</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Lipidomic Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Lipidomics Data/Lipidomics Visualization Dashboard.html">Lipidomics Visualization Dashboard</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Multi-omic Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Multi-omic Data/IntOmix.html">IntOmix</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Data Studio</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Data Studio/Data Studio.html">Introduction</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../Apps/Data Studio/CCLE Correlation Analysis.html">CCLE Correlation Analysis</a>
                </li>
    </ul>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Scaling Compute</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="Polly CLI.html">Polly CLI</a>
                </li>
                <li class=" current">
                    
    <a class="current" href="Polly Notebooks.html">Notebooks</a>
    <ul class="subnav">
            
    <li class="toctree-l3"><a href="#introduction">Introduction</a></li>
    
        <ul>
        
            <li><a class="toctree-l4" href="#pre-configured-environments">Pre-Configured Environments</a></li>
        
            <li><a class="toctree-l4" href="#computational-machines-available">Computational Machines Available</a></li>
        
            <li><a class="toctree-l4" href="#other-useful-features">Other Useful Features</a></li>
        
            <li><a class="toctree-l4" href="#getting-started-with-polly-notebook">Getting started with Polly Notebook</a></li>
        
            <li><a class="toctree-l4" href="#structure-of-polly-notebook">Structure of Polly Notebook</a></li>
        
            <li><a class="toctree-l4" href="#polly-offerings">Polly Offerings</a></li>
        
            <li><a class="toctree-l4" href="#accessing-workspace-files-in-notebook">Accessing Workspace files in Notebook</a></li>
        
            <li><a class="toctree-l4" href="#installing-packages">Installing Packages</a></li>
        
            <li><a class="toctree-l4" href="#reusable-scripts">Reusable Scripts</a></li>
        
        </ul>
    

    <li class="toctree-l3"><a href="#videos">Videos</a></li>
    

    </ul>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../Data Lake.html">Data Lake</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../Release Notes.html">Release Notes</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../index.html">Polly Documentation</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
    
      
        
          <li>Scaling Compute &raquo;</li>
        
      
    
    <li>Notebooks</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="introduction">Introduction</h1>
<p>Polly Notebook is a scalable analytics platform which allows you to perform data analysis remotely in a Jupyter-like notebook. It provides the flexibility to select the compute capacity, the environment according to your need along with the ability to share the analyses with your peers for seamless team collaboration.</p>
<p>Polly Notebook provides a Jupyter-like interface on the cloud. Some of the features of Polly Notebooks over other local hosting options are:</p>
<ul>
<li>
<p><strong>Ready-to-code platform:</strong> Installing and maintaining environments for every notebook can be a frustrating overhead. We provide custom docker environments that come pre-installed with modules commonly used in bioinformatics. You can also add your own custom docker environments.</p>
</li>
<li>
<p><strong>Cloud storage:</strong> With Polly Notebooks, you can store your data files and notebooks in a single place that will be ready to run in less than 5 minutes from anywhere in the world. No need to fetch your code from Bitbucket anymore!</p>
</li>
<li>
<p><strong>Share and collaborate on your Workspaces:</strong> Polly allows sharing of workspaces so you can review and refer notebooks within your team. </p>
</li>
<li>
<p><strong>Resource management:</strong> Most biological analyses (like RNAseq) are commonly resource-intensive, whether in terms of RAM or processing power. In such cases, you either have to scramble for bigger resources or compromise on the speed by using less processing power. Polly makes it possible to scale up your resources at any time.</p>
</li>
</ul>
<p><strong>Accessing Polly Notebooks</strong></p>
<p>Navigate to the <a href="https://docs.elucidata.io/Getting%20Started/Workspaces.html">Polly Workspaces</a> in which the analysis needs to be performed. The notebooks of this workspace can accessed from the middle panel.</p>
<p><img alt="Polly Workspaces" src="../img/Notebooks/Workspace.png" title="Polly Workspaces" /> <center><strong>Figure 1.</strong> Polly Workspaces</center></p>
<p>You can access Polly Notebooks in three ways:</p>
<ul>
<li><strong>Create a new notebook:</strong> Go to the <em>Applications</em> interface and click on the <em>Polly Notebooks</em> button located on the bottom left side of the navigation bar to create a new notebook. You are required to provide a name to the Notebook and select the workspace along with an environment and a machine to run the given notebook.</li>
</ul>
<p><img alt="New Notebook button to create a new notebook" src="../img/Notebooks/Create.png" /> <center><strong>Figure 2.</strong> <em>New Notebook</em> button to create a new notebook</center></p>
<ul>
<li><strong>Upload a notebook:</strong> Click on the <em>Cloud Upload icon</em> prsent at the top of the middle panel and select the <em>Upload a Notebook</em> option.</li>
</ul>
<p><img alt="Upload Notebook button to upload a notebook" src="../img/Notebooks/Notebookupload.png" /> <center><strong>Figure 3.</strong> <em>Upload Notebook</em> button to upload a notebook</center></p>
<p>The notebook can be uploaded from the local system as well as from the various cloud storage services (Dropbox, Google Drive and Box). To upload from local system, files can be dragged and dropped. To upload from various cloud storage services, select the relevant option, login to the service and select the files to be uploaded.</p>
<p><img alt="Window to import notebook from local or other cloud storage services" src="../img/Notebooks/Upload.png" /> <center><strong>Figure 4.</strong> Window to import notebook from local or other cloud storage services</center></p>
<ul>
<li><strong>Open an existing notebook:</strong> Click on the name of any existing notebook to <em>Edit and Launch</em> it.
If you are running the notebook for the first time, the option <em>Edit and Launch</em> would appear as a default selection to launch the selected notebook. You are required to select an environment and a machine to run the given notebook, oly after these selections are done you can launch the notebook.</li>
</ul>
<p><img alt="Opening a notebook" src="../img/Notebooks/Opennotebook.png" /><center><strong>Figure 5.</strong> Opening a notebook for the first time</center></p>
<ul>
<li><strong>For an older notebook:</strong> You have two options, you can either launch the notebook directly by the Launch button or you can choose to edit it first before launching through the Edit and Launch button. </li>
</ul>
<p><strong>Note:</strong></p>
<ul>
<li>Under the edit option, you can only change the machine type. The docker environment would remain the same as the one selected when you run the notebook for the first time.</li>
</ul>
<p><img alt="Opening a notebook" src="../img/Notebooks/Opennotebook2.png" /> <center><strong>Figure 6.</strong> Opening an old notebook</center></p>
<h2 id="pre-configured-environments">Pre-Configured Environments</h2>
<p>Polly supports various notebook environments in the form of dockers to cater to the needs of different users. Each of the dockers is built according to various data analytic needs ranging from basic scripting, processing large data or training and testing of ML models. The menu to select the notebook environments will pop-up whenever you create or upload the notebook and opens it for the first time.</p>
<p><img alt="Menu to select various available environments" src="../img/Notebooks/Env.png" /> <center><strong>Figure 7.</strong> Menu to select various available environments</center></p>
<p>The various notebook environments supported are as follows:</p>
<table>
<thead>
<tr>
<th>Environment</th>
<th>Usage</th>
<th>R libraries</th>
<th>Python Modules</th>
<th>System</th>
</tr>
</thead>
<tbody>
<tr>
<td>R</td>
<td>General R scripting</td>
<td><ul><li>askpass 1.1</li><li>assertthat 0.2.1</li><li>backports 1.1.5</li><li>base64enc 0.1-3</li><li>BH 1.72.0-3</li><li>BiocManager 1.30.10</li><li>bitops 1.0-6</li><li>brew 1.0-6</li><li>callr 3.4.2</li><li>cli 2.0.2</li><li>clipr 0.7.0</li><li>clisymbols 1.2.0</li><li>colorspace 1.4-1</li><li>commonmark 1.7</li><li>covr 3.4.0</li><li>crayon 1.3.4</li><li>crosstalk 1.0.0</li><li>curl 4.3</li><li>cyclocomp 1.1.0</li><li>desc 1.2.0</li><li>devtools 2.2.2</li><li>diffobj 0.2.3</li><li>digest 0.6.25</li><li>DT 0.12</li><li>ellipsis 0.3.0</li><li>evaluate 0.14</li><li>fansi 0.4.1</li><li>farver 2.0.3</li><li>fastmap 1.0.1</li><li>foghorn 1.1.4</li><li>fs 1.3.1</li><li>gargle 0.4.0</li><li>ggplot2 3.2.1</li><li>gh 1.1.0</li><li>git2r 0.26.1</li><li>glue 1.3.1</li><li>gmailr 1.0.0</li><li>gridExtra 2.3</li><li>gtable 0.3.0</li><li>highlight 0.5.0</li><li>highr 0.8</li><li>htmltools 0.4.0</li><li>htmlwidgets 1.5.1</li><li>httpuv 1.5.2</li><li>httr 1.4.1</li><li>hunspell 3.0</li><li>ini 0.3.1</li><li>IRdisplay 0.7.0</li><li>IRkernel 1.1</li><li>jsonlite 1.6.1</li><li>knitr 1.28</li><li>labeling 0.3</li><li>later 1.0.0</li><li>lazyeval 0.2.2</li><li>leaflet 2.0.3</li><li>leaflet.providers 1.9.0</li><li>lifecycle 0.1.0</li><li>lintr 2.0.1</li><li>magrittr 1.5</li><li>markdown 1.1</li><li>memoise 1.1.0</li><li>mime 0.9</li><li>mockery 0.4.2</li><li>munsell 0.5.0</li><li>openssl 1.4.1</li><li>parsedate 1.2.0</li><li>pbdZMQ 0.3-3</li><li>pillar 1.4.3</li><li>pingr 2.0.0</li><li>pkgbuild 1.0.6</li><li>pkgconfig 2.0.3</li><li>pkgdown 1.4.1</li><li>pkgload 1.0.2</li><li>plyr 1.8.6</li><li>png 0.1-7</li><li>PollyConnector 0.0.0</li><li>praise 1.0.0</li><li>prettyunits 1.1.1</li><li>processx 3.4.2</li><li>promises 1.1.0</li><li>ps 1.3.2</li><li>purrr 0.3.3</li><li>R6 2.4.1</li><li>rappdirs 0.3.1</li><li>raster 3.0-12</li><li>rcmdcheck 1.3.3</li><li>RColorBrewer 1.1-2</li><li>Rcpp 1.0.3</li><li>rematch 1.0.1</li><li>rematch2 2.1.0</li><li>remotes 2.1.1</li><li>repr 1.1.0</li><li>reshape2 1.4.3</li><li>reticulate 1.14</li><li>rex 1.1.2</li><li>rhub 1.1.1</li><li>RJSONIO 1.3-1.4</li><li>rlang 0.4.5</li><li>rmarkdown 2.1</li><li>roxygen2 7.0.2</li><li>rprojroot 1.3-2</li><li>rstudioapi 0.11</li><li>rversions 2.0.1</li><li>rvest 0.3.5</li><li>scales 1.1.0</li><li>selectr 0.4-2</li><li>sessioninfo 1.1.1</li><li>shiny 1.4.0</li><li>sourcetools 0.1.7</li><li>sp 1.4-1</li><li>spelling 2.1</li><li>stringi 1.4.6</li><li>stringr 1.4.0</li><li>sys 3.3</li><li>testthat 2.3.2</li><li>tibble 2.1.3</li><li>tinytex 0.20</li><li>triebeard 0.3.0</li><li>urltools 1.7.3</li><li>usethis 1.5.1</li><li>utf8 1.1.4</li><li>uuid 0.1-4</li><li>vctrs 0.2.3</li><li>viridis 0.5.1</li><li>viridisLite 0.3.0</li><li>whisker 0.4</li><li>whoami 1.3.0</li><li>withr 2.1.2</li><li>xfun 0.12</li><li>xml2 1.2.2</li><li>xmlparsedata 1.0.3</li><li>xopen 1.0.0</li><li>xtable 1.8-4</li><li>yaml 2.2.1</li><ul></td>
<td>None</td>
<td></td>
</tr>
<tr>
<td>Python 2</td>
<td>General Python 2 scripting</td>
<td>None</td>
<td><ul><li>attrs 19.3.0</li><li>backports-abc 0.5</li><li>backports.functools-lru-cache 1.6.1</li><li>backports.shutil-get-terminal-size 1.0.0</li><li>bleach 3.1.0</li><li>certifi 2019.11.28</li><li>chardet 3.0.4</li><li>cmapPy 1.0.5</li><li>configparser 4.0.2</li><li>contextlib2 0.6.0.post1</li><li>cycler 0.10.0</li><li>decorator 4.4.1</li><li>defusedxml 0.6.0</li><li>entrypoints 0.3</li><li>enum34 1.1.6</li><li>funcsigs 1.0.2</li><li>functools32 3.2.3.post2</li><li>futures 3.3.0</li><li>h5py 2.10.0</li><li>idna 2.8</li><li>importlib-metadata 1.5.0</li><li>ipaddress 1.0.23</li><li>ipykernel 4.10.1</li><li>ipython 5.9.0</li><li>ipython-genutils 0.2.0</li><li>ipywidgets 7.4.2</li><li>Jinja2 2.11.1</li><li>jsonschema 3.2.0</li><li>jupyter-client 5.3.4</li><li>jupyter-core 4.6.2</li><li>kiwisolver 1.1.0</li><li>MarkupSafe 1.1.1</li><li>matplotlib 2.2.4</li><li>mistune 0.8.4</li><li>nbconvert 5.6.1</li><li>nbformat 4.4.0</li><li>notebook 5.7.8</li><li>numpy 1.16.6</li><li>pandas 0.24.1</li><li>pandocfilters 1.4.2</li><li>pathlib2 2.3.5</li><li>pexpect 4.8.0</li><li>pickleshare 0.7.5</li><li>plotly 3.7.0</li><li>prometheus-client 0.7.1</li><li>prompt-toolkit 1.0.18</li><li>ptyprocess 0.6.0</li><li>PubChemPy 1.0.4</li><li>Pygments 2.5.2</li><li>pyparsing 2.4.6</li><li>pyrsistent 0.15.7</li><li>python-dateutil 2.8.1</li><li>pytz 2019.3</li><li>pyzmq 18.1.1</li><li>qgrid 1.1.1</li><li>requests 2.21.0</li><li>retrying 1.3.3</li><li>scandir 1.10.0</li><li>scikit-learn 0.20.3</li><li>scipy 1.2.3</li><li>Send2Trash 1.5.0</li><li>simplegeneric 0.8.1</li><li>singledispatch 3.4.0.3</li><li>six 1.14.0</li><li>subprocess32 3.5.4</li><li>terminado 0.8.3</li><li>testpath 0.4.4</li><li>tornado 5.1.1</li><li>traitlets 4.3.3</li><li>urllib3 1.24.3</li><li>wcwidth 0.1.8</li><li>webencodings 0.5.1</li><li>widgetsnbextension 3.4.2</li><li>zipp 1.1.0</li><ul></td>
<td></td>
</tr>
<tr>
<td>Python 3</td>
<td>General Python 3 scripting</td>
<td>None</td>
<td><ul><li>alembic 1.4.1</li><li>async-generator 1.10</li><li>attrs 19.3.0</li><li>awscli 1.17.12</li><li>backcall 0.1.0</li><li>bleach 3.1.1</li><li>botocore 1.14.12</li><li>certifi 2019.11.28</li><li>chardet 3.0.4</li><li>colorama 0.4.3</li><li>cycler 0.10.0</li><li>decorator 4.4.2</li><li>defusedxml 0.6.0</li><li>docutils 0.15.2</li><li>entrypoints 0.3</li><li>idna 2.8</li><li>importlib-metadata 1.5.0</li><li>ipykernel 5.1.4</li><li>ipython 7.13.0</li><li>ipython-genutils 0.2.0</li><li>ipywidgets 7.5.1</li><li>jedi 0.16.0</li><li>Jinja2 2.11.1</li><li>jmespath 0.9.5</li><li>jsonschema 3.2.0</li><li>jupyter-client 6.0.0</li><li>jupyter-core 4.6.3</li><li>jupyter-dashboards 0.7.0</li><li>jupyterhub 0.9.4</li><li>kiwisolver 1.1.0</li><li>Mako 1.1.2</li><li>MarkupSafe 1.1.1</li><li>matplotlib 2.2.3</li><li>mistune 0.8.4</li><li>nbconvert 5.6.1</li><li>nbformat 5.0.4</li><li>notebook 5.7.2</li><li>numpy 1.18.1</li><li>pamela 1.0.0</li><li>pandas 1.0.1</li><li>pandocfilters 1.4.2</li><li>parso 0.6.2</li><li>pexpect 4.8.0</li><li>pickleshare 0.7.5</li><li>prometheus-client 0.7.1</li><li>prompt-toolkit 3.0.3</li><li>ptyprocess 0.6.0</li><li>pyasn1 0.4.8</li><li>Pygments 2.5.2</li><li>pyparsing 2.4.6</li><li>pyrsistent 0.15.7</li><li>python-dateutil 2.8.1</li><li>python-editor 1.0.4</li><li>python-oauth2 1.1.1</li><li>pytz 2019.3</li><li>PyYAML 5.3</li><li>pyzmq 19.0.0</li><li>qgrid 1.3.0</li><li>requests 2.21.0</li><li>rsa 3.4.2</li><li>s3transfer 0.3.3</li><li>Send2Trash 1.5.0</li><li>six 1.14.0</li><li>SQLAlchemy 1.3.13</li><li>terminado 0.8.3</li><li>testpath 0.4.4</li><li>tornado 5.1.1</li><li>traitlets 4.3.3</li><li>urllib3 1.24.3</li><li>wcwidth 0.1.8</li><li>webencodings 0.5.1</li><li>widgetsnbextension 3.5.1</li><li>zipp 3.1.0</li><ul></td>
<td></td>
</tr>
<tr>
<td>Pollyglot</td>
<td>Multiple kernels (R, python and bash) in same notebook/environment</td>
<td><ul><li>All libraries from base R docker</li><li>Seurat</li><li>pagoda2</li><li>CellRanger</li><li>SingleR</li><ul></td>
<td><ul><li>All libraries from base python docker</li><li>scanPy</li><li>velocyto</li><li><a href="https://github.com/YosefLab/scVI">scVI</a>(scVI supports pytorch)</li><li>louvain</li><ul></td>
<td></td>
</tr>
<tr>
<td>Barcoded Bulk RNA-seq</td>
<td>Alignment and processing of RNA-seq fastq files with barcodes</td>
<td><ul><li>All libraries from R docker</li><li>limma</li><li>affy</li><li>DESeq2</li><li>edgeR</li><li>cqn</li><li>sva</li><li>BioMart</li><li>mygene</li><li>amritr</li><li>Boruta</li><li>fgsea</li><li>gsva</li><li>ReactomePA</li><li>xCell</li><li>singleR</li><li>enrichR</li><li>org.Hs.eg.db</li><li>org.Mm.eg.db</li><li>Annotation dbi</li><li>clusterProfiler</li><ul></td>
<td>&#xfeff;</td>
<td><ul><li>STARsubread-1.6.4-source</li><li>gosaamer</li><li>Fastqc</li><li>Multiqc</li><li>Picard</li><ul></td>
</tr>
<tr>
<td>Machine Learning in python</td>
<td>Training, testing and validation of ML models</td>
<td>None</td>
<td><ul><li>All libraries from base python docker</li><li>h5py</li><li>keras</li><li>lightgbm</li><li>tensorflow</li><li>xgboost</li><ul></td>
<td></td>
</tr>
<tr>
<td>Single Cell Downstream</td>
<td>Single Cell Analysis</td>
<td><ul><li>All libraries from base R docker</li><li>Seurat</li><li>pagoda2</li><li>CellRanger</li><li>SingleR</li><li>ExperimentHub</li><ul></td>
<td><ul><li>All libraries from base python docker</li><li>scanPy</li><li>velocyto</li><li><a href="https://github.com/YosefLab/scVI">scVI</a>(scVI supports pytorch)</li><li>louvain</li><li>rpy2</li><li>anndata2ri</li><ul></td>
<td></td>
</tr>
<tr>
<td>Data Exploration</td>
<td>R and python for general data analysis</td>
<td>All libraries from base R docker</td>
<td>All libraries from base python docker</td>
<td></td>
</tr>
<tr>
<td>RNA-seq Downstream</td>
<td>Transcriptomics Analysis</td>
<td><ul><li>All libraries from R docker</li><li>limma</li><li>affy</li><li>DESeq2</li><li>edgeR</li><li>cqn</li><li>sva</li><li>BioMart</li><li>mygene</li><li>amritr</li><li>Boruta</li><li>fgsea</li><li>gsva</li><li>ReactomePA</li><li>xCell</li><li>singleR</li><li>enrichR</li><li>org.Hs.eg.db</li><li>org.Mm.eg.db</li><li>Annotation dbi</li><li>clusterProfiler</li><ul></td>
<td>All libraries from base python docker</td>
<td></td>
</tr>
<tr>
<td>Metabolomics</td>
<td>Metabolomics Analysis</td>
<td><ul><li><a href="https://github.com/sneumann/CAMERA, https://www.bioconductor.org/packages/release/bioc/html/CAMERA.html">CAMERA</a></li><li><a href="https://www.bioconductor.org/packages/release/bioc/html/xcms.html">XCMS</a></li><li>limma</li><li>matrixStats</li><li>stats</li><li>stringr</li><li>ggplot2</li><li>plotly</li><li>ggsci</li><li>latex2exp</li><li>dplyr</li><li>ggrepel</li><li>SuperExactTest</li><li>UpSetR</li><li><a href="https://bitbucket.org/elucidatainc/pollycommonr/src/master/PollyCommonR/">PollyCommonR</a></li><li>mapGCT</li><li>MetaboAnalystR</li><li>KEGGREST</li><li><a href="https://www.bioconductor.org/packages/release/bioc/html/pathview.html">Pathview</a></li><li>KEGGgraph</li><li><a href="https://github.com/cmap/morpheus.R">Morpheus</a></li><li>Pca3d</li><li>X13CMS</li><li>Phantasus</li><li>MSnbase</li><li>MAIT</li><li>cmapR</li><li>xMSannotator</li><ul></td>
<td>All libraries from base python docker</td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="computational-machines-available">Computational Machines Available</h2>
<p>The size of the data varies from few MBs to hundreds of GBs, and in order to process and analyze this huge data, one would need the computation power from a small machine to a large workstation. Polly Notebook supports configurations having 2 to 72 GB Ram and 1 to 36 CPU cores. The menu to select a machine configuration will pop-up when you creates a new notebook or uploads a notebook and tries to open it for the first time.</p>
<p><img alt="Menu to select various machine configurations" src="../img/Notebooks/Machine.png" /> <center><strong>Figure 8.</strong> Menu to select various machine configurations</center></p>
<p>Most of the machine configuration are already specified to cover the wide variety of use cases. More machine configuration can also be made available on request (contact us at <a href="polly@elucidata.io">polly@elucidata.io</a>). The general machine configurations are divided into three broad categories:</p>
<ul>
<li><strong>General purpose:</strong> Configurations from 1 to 4 CPU cores and 2 to 16 GB RAM fall under this category. The various configurations are:</li>
</ul>
<table>
<thead>
<tr>
<th>Name</th>
<th>CPU/Cores</th>
<th>RAM</th>
</tr>
</thead>
<tbody>
<tr>
<td>Polly small</td>
<td>1</td>
<td>2 GB</td>
</tr>
<tr>
<td>Polly medium</td>
<td>2</td>
<td>4 GB</td>
</tr>
<tr>
<td>Polly large</td>
<td>2</td>
<td>8 GB</td>
</tr>
<tr>
<td>Polly x-large</td>
<td>4</td>
<td>16 GB</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>Compute Intensive:</strong> Configurations from 16 to 36 CPU cores and 32 to 72 GB RAM fall under this category. The various configurations are:</li>
</ul>
<table>
<thead>
<tr>
<th>Name</th>
<th>CPU/Cores</th>
<th>RAM</th>
</tr>
</thead>
<tbody>
<tr>
<td>Polly 2x-large</td>
<td>16</td>
<td>32 GB</td>
</tr>
<tr>
<td>Polly 3x-large</td>
<td>36</td>
<td>72 GB</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>Memory-Optimized:</strong> Configurations from 4 to 8 CPU cores and 32 to 64 GB RAM fall under this category. The various configurations are:</li>
</ul>
<table>
<thead>
<tr>
<th>Name</th>
<th>CPU/Cores</th>
<th>RAM</th>
</tr>
</thead>
<tbody>
<tr>
<td>Polly 2x-large</td>
<td>4</td>
<td>32 GB</td>
</tr>
<tr>
<td>Polly 3x-large</td>
<td>8</td>
<td>64 GB</td>
</tr>
<tr>
<td>Polly 4x-large</td>
<td>16</td>
<td>120GB</td>
</tr>
</tbody>
</table>
<h2 id="other-useful-features">Other Useful Features</h2>
<p>There are few other useful features as well that might come handy when using a Polly Notebook. </p>
<ul>
<li>
<p>Click on the kebab menu at the end of the selected notebook. A menu with various options will open.</p>
<p><img alt="Edit* button to change machine configuration" src="../img/Notebooks/Features.png" /> <center><strong>Figure 9.</strong> <em>Edit</em> button to change machine configuration</center></p>
</li>
<li>
<p>Scroll down in the menu and navigate to the desired option.</p>
<ul>
<li>
<p><em>Rename</em>: You can rename the file using this option. Provide the new name to the file and click on <em>Rename</em> to confirm your changes.   </p>
</li>
<li>
<p><em>Edit</em>: Polly gives the flexibility to change the machine configuration to allow the usage of the notebook according to the computing power required at each step. You can change the configuration according to the need at each step.
A menu with the different machine configuration will open, with the various options available will be displayed under the <em>Select Machine Type</em> segment. Select the appropriate option to change the configuration.</p>
</li>
<li>
<p><em>Download</em>: You can select this option to download the selected notebook on your system.</p>
</li>
<li>
<p><em>Delete</em>: You can use this option to delete the selected notebook. </p>
</li>
</ul>
</li>
</ul>
<h2 id="getting-started-with-polly-notebook">Getting started with Polly Notebook</h2>
<p>Upon selecting a pre-configured docker environment and a computational machine, a Polly Notebook starts launching on a new tab of the browser. Based upon the type of computational machine chosen while launching a Polly Notebook you will see a progress bar which will tell you that your new notebook is opening.</p>
<p><img alt="Progress bar upon launching a Polly Notebook" src="../img/Notebooks/16.png" /> <center><strong>Figure 10.</strong> Progress bar upon launching a Polly Notebook</center></p>
<p>Once the server is ready, you will see the new notebook gets opened on the browser. The interface is very similar to that of a Jupyter notebook.</p>
<p><img alt="Polly Notebook interface" src="../img/Notebooks/17.png" /> <center><strong>Figure 11.</strong> Polly Notebook interface</center></p>
<p>On the top left, you can see a pre-defined name given to the notebook if in case a new notebook was created. Towards the top right, you can see the Polly Workspace name and below it, you can see the kernel/docker environment selected for opening the notebook.</p>
<ul>
<li>
<p><strong>Menu bar:</strong> There are multiple tabs present in the menu bar section which can be used to operate various functions in the notebook. For example, under the <em>File</em> tab, you can select the Rename option to change the name of the current active notebook.</p>
</li>
<li>
<p><strong>Toolbar:</strong>  It contains multiple icons that allow you to perform various operations that are frequently used.</p>
</li>
</ul>
<h2 id="structure-of-polly-notebook">Structure of Polly Notebook</h2>
<p>The Polly notebook comprises of a sequence of cells. There are three types of cells: <strong>markdown cells</strong>, <strong>raw cells</strong>, and <strong>code cells</strong>. In each of these types, you can input multi-line content and each cell can be executed by pressing <em>Shift+Enter</em>, or by clicking either the <em>Run cells</em> option on <em>Cell</em> tab in the menu bar or the “Play” button in the toolbar.</p>
<p><img alt="Structure of a Polly Notebook" src="../img/Notebooks/18.png" /> <center><strong>Figure 12.</strong> Structure of a Polly Notebook</center></p>
<p><strong>Markdown cells</strong></p>
<p>You can record the computational process in a proficient manner using <em>rich text</em>. The Markdown language allows you to define a structure to the notebook by using markdown headings. It gives a basic method to play out text markup, that is, to determine which parts of the text should be stressed (italics), bold, form lists, etc.</p>
<p><strong>Raw cells</strong></p>
<p>You can write output directly in the <em>raw cells</em>. A raw cell is not evaluated by a notebook meaning anything written in the raw cell goes to the output when that cell is executed.</p>
<p><strong>Code cells</strong></p>
<p>A <em>code cell</em> allows you to edit and write a new code. The <em>code cell</em> executes the code written by you based on the kernel selected while launching the notebook. The <em>code cell</em> can include multiple programming languages as well as seen on the bottom right side of the image above. The above example is of a Pollyglot Docker environment which allows you to select multiple programming languages in the same notebook thus, you can select the type of kernel you prefer to code on.</p>
<p>Once the <em>code cell</em> is executed, the results which are computed by sending the code to the kernel are displayed as an output below the cell. Again to execute a <em>code cell</em>, you can click on the “Run” button and if you want to stop the computation process of a particular <em>code cell</em>, then the “Interrupt” button needs to be selected in the toolbar.</p>
<p><img alt="Running a code cell" src="../img/Notebooks/19.png" /> <center><strong>Figure 13.</strong> Running a code cell</center></p>
<h2 id="polly-offerings">Polly Offerings</h2>
<p>Polly Offerings tab in the Menu bar contains the following two options, namely <em>Terminal</em> and <em>File Explorer</em> which are described below.</p>
<p><img alt="Polly Offerings tab" src="../img/Notebooks/20.png" /> <center><strong>Figure 14.</strong> Polly Offerings tab</center></p>
<p><strong>Terminal</strong></p>
<p>Once the <em>Terminal</em> option is selected, it launches a new tab on the browser and provides access to the command-line interface to execute any sets of commands. You have access to all the file types which are available in the docker environment and those can be managed through the terminal as well. The terminal option also allows you to install Python or R packages (as described later), managing system binaries and system configurations, and helps you working with code repositories hosted on GitHub, Bitbucket, etc.</p>
<p><img alt="Terminal screen window" src="../img/Notebooks/21.png" /> <center><strong>Figure 15.</strong> Terminal screen window</center></p>
<p><strong>File Explorer</strong></p>
<p>Similar to the above option, if you select the <em>File Explorer</em> option, a new tab opens up in the browser and you can view different file types and directories present in the docker environment. Under the <em>Files</em> tab, the list of all the files and directories is available to you and any modification such as delete, upload or modifying by opening a file type can be done.</p>
<p><img alt="Demo Data for FirstView" src="../img/Notebooks/22.png" /> <center><strong>Figure 16.</strong> File Explorer window</center></p>
<p>Additionally, you can also launch a new notebook by selecting the <em>New</em> button present on the top right corner of the page in <em>File Explorer</em>. The new notebook will open in a new tab and would automatically be made available in the Notebook section of the same Polly Workspace of the original notebook.</p>
<p><img alt="Launching a new notebook using File Explorer" src="../img/Notebooks/23.png" /> <center><strong>Figure 17.</strong> Launching a new notebook using <em>File Explorer</em></center></p>
<p><em>File Explorer</em> window also allows you to view, edit or create various file types in an interactive manner. The <em>Text File</em> option in the New button can be used to create a new text file. For viewing or editing a file, you can click on the file and a text editor will open in a new tab of the browser. You can view or edit the file and save the changes made in the file. The text editor also allows you to select a programming language from the Language tab to edit and convert the file format.</p>
<p><img alt="Opening a file using a Text editor" src="../img/Notebooks/Languages.jpeg" /> <center><strong>Figure 18.</strong> Opening a file using a Text editor</center></p>
<h2 id="accessing-workspace-files-in-notebook">Accessing Workspace files in Notebook</h2>
<p><strong>Accessing individual files using Python and R functions</strong></p>
<p>For carrying on analysis, if you require any input files which are available in Polly Workspaces, those files can be fetched using a set of commands. You can list all the files present in the Workspace and then select the individual file by the following command:</p>
<pre><code>## Lists all the files present in the project
list_project_file()
## The file will be downloaded in the current working directory
download_project_file('sample_file.csv')
</code></pre>

<p>After finishing the analysis, you can push back the newly generated output files again to the Workspace using the following command</p>
<pre><code>## Save the file to the project
save_file_to_project('sample_file.csv')
</code></pre>

<p><img alt="Accessing individual files in a notebook" src="../img/Notebooks/25.png" /> <center><strong>Figure 19.</strong> Accessing individual files in a notebook</center></p>
<p><strong>Note:</strong></p>
<ul>
<li>These functions cannot access files within folders in workspace. To access those files, use CLI commands.</li>
</ul>
<p><strong>Accessing files and directories using CLI commands</strong></p>
<p>The contents of any directory within a Workspace can be listed using the following command on a notebook terminal or a bash cell.</p>
<pre><code>polly files list --workspace-path "<path/to/the/directory>" -y
</code></pre>

<p>Here, the path of the directory has to start with “polly://”. To view the contents within a folder called “Data” in the workspace, the following command will have to be executed on the notebook terminal.</p>
<pre><code>polly files list --workspace-path "polly://Data" -y
</code></pre>

<p>To access the directory in the notebook, the following command will have to be executed on the notebook terminal or a bash cell.</p>
<pre><code>polly files sync -s "<path/to/the/project/directory>" -d "<absolute/path/to/the/notebook/directory>" -y
</code></pre>

<p>Here, <code>-s</code> refers to source and <code>-d</code> refers to destination. If the folder called “Data” is to be accessed from Workspace in the notebook folder called “Input”, execute the following command.</p>
<pre><code>polly files sync -s "polly://Data" -d "Input" -y
</code></pre>

<p>To save notebook directories back to the Workspace, keep the source as notebook directory and destination as Polly Workspace in the same command as mentioned above.</p>
<pre><code>polly files sync -s "<absolute/path/to/the/notebook/directory>" -d "<path/to/the/project/directory>" -y
</code></pre>

<p>To save the folder called “Output” back to Polly Workspace, use the following command.</p>
<pre><code>polly files sync -s "Output" -d "polly://" -y</code></pre>

<p>Similarly, if an individual file needs to be accessed in a notebook, use the following command</p>
<pre><code>polly files copy -s "" -d "" -y
</code></pre>

<p>Here, <code>-s</code> refers to source and <code>-d</code> refers to destination. If the file called “Input1.csv” is to be accessed from Workspace folder “Data” in the notebook folder called “Input”, execute the following command.</p>
<pre><code>polly files copy -s "polly://Data/Input1" -d "Input/Input1.csv" -y</code></pre>

<p>An individual file can be saved back to workspace by interchanging source and destination in the mentioned command. </p>
<pre><code>polly files copy -s "Input/Input1.csv" -d "polly://Data/Input1" -y</code></pre>

<h2 id="installing-packages">Installing Packages</h2>
<p>Although most of the required packages and tools can be made available to you via the customized docker environment, sometimes you might require to install new packages to carry on the analysis. For installing the packages, you can choose two options based on their convenience, you can do it on the Notebook itself or via the terminal.</p>
<p><strong>Installing packages and system binaries using the Notebook cell</strong></p>
<p>You can install the required packages and system binaries by running the usual installation codes on the code cell of a notebook.</p>
<ul>
<li><strong>For Python packages:</strong> You can run the following command in the code cell with Python kernel selected to install the required packages.</li>
</ul>
<pre><code># for installing packages DON'T forget to use sudo. It will not ask for password.
!sudo pip install <package-name>
</code></pre>

<ul>
<li><strong>For R packages:</strong> You can run the following command in the code cell with R kernel selected to install the required packages.</li>
</ul>
<pre><code># for installing packages DON'T forget to use sudo. It will not ask for password.
## Installing CRAN packages
!sudo  R -e 'install.packages(c("package-name"), repos="https://cloud.r-project.org/")'

## Installing Bioconductor packages
!sudo R -e 'BiocManager::install(c("package-name"), update = TRUE, ask = FALSE)'
# If error finding BiocManager then install it first using the following command and re-run the above command.
!sudo  R -e 'install.packages(c("BiocManager"), repos="https://cloud.r-project.org/")'
</code></pre>

<p><img alt="Installing R and Python packages" src="../img/Notebooks/26.png" /> <center><strong>Figure 20.</strong> Installing R and Python packages</center></p>
<ul>
<li><strong>For System binaries:</strong> You can also install the system binaries by running the following command in the <em>code cell</em> selecting the <em>bash</em> kernel.</li>
</ul>
<pre><code># System binaries
sudo apt install <package-name>

# If the above command outputs package not found, You can run this command to update the system package indices
sudo apt-get update
</code></pre>

<p><img alt="Installing System binaries using the Notebook code cell" src="../img/Notebooks/27.png" /> <center><strong>Figure 21.</strong> Installing System binaries using the Notebook <em>code cell</em></center></p>
<p><strong>Installing packages and system binaries via Terminal</strong></p>
<p>Another option is also available to install various packages and system binaries using the terminal. You can access the terminal as described in the document above. The commands for installation are almost similar to commands used while installing using a notebook <em>code cell</em>.</p>
<ul>
<li><strong>For Python packages:</strong> You can run the following command directly on the terminal to install the required packages. Once the package installation is successful, you can import the package in your notebook.</li>
</ul>
<pre><code># for installing packages DON'T forget to use sudo. It will not ask for password.
> sudo pip install <package-name>
</code></pre>

<p><img alt="Installing Python packages using the Terminal" src="../img/Notebooks/28.png" /> <center><strong>Figure 22.</strong> Installing Python packages using the Terminal</center></p>
<ul>
<li><strong>For R packages:</strong> You are required to go to the terminal and open the R Kernel using “sudo R” and then install the required R packages. Once the package installation is successful, you can import the library in your notebook  R kernel as usual.</li>
</ul>
<pre><code>## You can install R package by opening R terminal
> sudo R

## Install CRAN packages using the following command
> install.packages(c('pkg-name'), dependencies=TRUE, repos=<Enter your choice cran mirror link>)
# For cran mirror link: You can use either of your choice or this one : "https://cran.cnr.berkeley.edu/"

## Install Bioconductor packages using the following command
> BiocManager::install(c("pkg-name"), update = TRUE, ask = FALSE)
# If error finding BiocManager then install it first using the following command and re-run the above command.
> install.packages("BiocManager")
</code></pre>

<p><img alt="Installing R packages using the Terminal" src="../img/Notebooks/29.png" /> <center><strong>Figure 23.</strong> Installing R packages using the Terminal</center></p>
<ul>
<li><strong>For System binaries:</strong> You can also install the system binaries by running the following command directly on the terminal itself.</li>
</ul>
<pre><code># System binaries
> sudo apt install <package-name>

# If the above command outputs package not found, You can run this command to update the system package indices
> sudo apt-get update
</code></pre>

<p><img alt="Installing System libraries using the Terminal" src="../img/Notebooks/30.png" /> <center><strong>Figure 24.</strong> Installing System libraries using the Terminal</center></p>
<h2 id="reusable-scripts">Reusable Scripts</h2>
<p>Polly Notebook also allows you to make use of the reusable scripts which are already made available to you in every notebook. The reusable scripts consist of the snippet codes which are required frequently to perform any analysis. The scripts can include data reading, normalization, visualization generic functions/codes and can be added to the notebook <em>code cell</em> with just a single click and executed as usual.  The reusable scripts can be found on the left side as a collapsible dialogue box and you can choose the scripts at any time while performing the analysis.</p>
<p><img alt="Reusable scripts on Polly Notebook" src="../img/Notebooks/31.png" /> <center><strong>Figure 25.</strong> Reusable scripts on Polly Notebook</center></p>
<p>On the right side, another collapsible dialogue box gets opened when you select any reusable script which provides information about the options and usage of that particular reusable script. You can also add your own reusable scripts on the Polly Notebook so as to make use of them in your repeated analysis and save time.</p>
<p><img alt="Options and Information of Reusable scripts" src="../img/Notebooks/32.png" /> <center><strong>Figure 26.</strong> Options and Information of Reusable scripts</center></p>
<h1 id="videos">Videos</h1>
<div style="position: relative; padding-bottom: 56.25%; height: 0; overflow: hidden; max-width: 100%; height: auto;">
    <iframe src="https://www.youtube.com/embed/jdNeaXtzctE" frameborder="0" allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;"></iframe>
</div>

<p><br></p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../Data Lake.html" class="btn btn-neutral float-right" title="Data Lake">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="Polly CLI.html" class="btn btn-neutral" title="Polly CLI"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="Polly CLI.html" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../Data Lake.html" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
