<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="img/elucidata.ico">
  
  <title>Data Lake - Polly Documentation</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="css/theme.css" type="text/css" />
  <link rel="stylesheet" href="css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Data Lake";
    var mkdocs_page_input_path = "Data Lake.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="js/jquery-2.1.1.min.js" defer></script>
  <script src="js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href="index.html" class="icon icon-home"> Polly Documentation</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="./search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="index.html">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Getting Started</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="Getting Started/Workspaces.html">Workspaces</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Apps</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="Apps/Introduction.html">Introduction</a>
                </li>
                <li class="">
                    
    <a class="" href="Apps/Host Apps.html">Host Apps</a>
                </li>
                <li class="">
                    
    <span class="caption-text">Metabolomic Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Metabolomic Data/El-MAVEN Phi Relative LC-MS.html">El-MAVEN Phi Relative LC-MS</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Metabolomic Data/MetScape.html">MetScape</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Metabolomic Data/FirstView.html">FirstView</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Metabolomic Data/Untargeted Pipeline.html">Untargeted Pipeline</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Metabolomic Data/Dual Mode Visualization.html">Dual Mode Data Visualization</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Metabolomic Data/QuantFit.html">QuantFit</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Metabolomic Data/El-MAVEN.html">El-MAVEN</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Metabolomic Data/Labeled LC-MSMS Workflow.html">Labeled LC-MS/MS Workfow</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Metabolomic Data/Labeled LC-MS Workflow.html">Labeled LC-MS Workflow</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Metabolomic Data/CompoundDiscoverer QuantFit.html">Compound Discoverer<sup>TM</sup> QuantFit</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Screening Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Screening Data/CRISPR Screening.html">CRISPR Screening</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Screening Data/High Throughput Drug Screening.html">High Throughput Drug Screening</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Sequencing Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Sequencing Data/RNA Seq Workflow.html">RNA Seq Workflow</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Proteomic Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Proteomic Data/Proteomics Workflow.html">Proteomics Workflow</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Lipidomic Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Lipidomics Data/Lipidomics Visualization Dashboard.html">Lipidomics Visualization Dashboard</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Multi-omic Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Multi-omic Data/IntOmix.html">IntOmix</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Data Studio</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Data Studio/Data Studio.html">Introduction</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Apps/Data Studio/CCLE Correlation Analysis.html">CCLE Correlation Analysis</a>
                </li>
    </ul>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Scaling Compute</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="Scaling compute/Polly CLI.html">Polly CLI</a>
                </li>
                <li class="">
                    
    <a class="" href="Scaling compute/Polly Notebooks.html">Notebooks</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1 current">
		
    <a class="current" href="Data Lake.html">Data Lake</a>
    <ul class="subnav">
            
    <li class="toctree-l2"><a href="#introduction">Introduction</a></li>
    
        <ul>
        
            <li><a class="toctree-l3" href="#available-public-data-repositories">Available public data repositories</a></li>
        
        </ul>
    

    <li class="toctree-l2"><a href="#dataset-filtering">Dataset Filtering</a></li>
    
        <ul>
        
            <li><a class="toctree-l3" href="#dataset-filtering-dashboard">Dataset Filtering Dashboard</a></li>
        
        </ul>
    

    <li class="toctree-l2"><a href="#data-lake-applications">Data Lake Applications</a></li>
    

    <li class="toctree-l2"><a href="#polly-discover-application">Polly Discover Application</a></li>
    
        <ul>
        
            <li><a class="toctree-l3" href="#gtex">GTEx</a></li>
        
            <li><a class="toctree-l3" href="#single-cell-visualization">Single Cell Visualization</a></li>
        
        </ul>
    

    <li class="toctree-l2"><a href="#access-through-polly-notebook-interface">Access through Polly Notebook interface</a></li>
    
        <ul>
        
            <li><a class="toctree-l3" href="#structure-of-a-data-repository">Structure of a data repository</a></li>
        
            <li><a class="toctree-l3" href="#usage">Usage</a></li>
        
            <li><a class="toctree-l3" href="#querying-at-the-dataset-level">Querying at the dataset level</a></li>
        
            <li><a class="toctree-l3" href="#querying-at-the-sample-level">Querying at the sample level</a></li>
        
            <li><a class="toctree-l3" href="#querying-at-the-feature-level">Querying at the feature level</a></li>
        
            <li><a class="toctree-l3" href="#access-annotation-repositories">Access annotation repositories</a></li>
        
            <li><a class="toctree-l3" href="#advanced-queries">Advanced queries</a></li>
        
            <li><a class="toctree-l3" href="#downloading-datasets">Downloading datasets</a></li>
        
        </ul>
    

    <li class="toctree-l2"><a href="#videos">Videos</a></li>
    

    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="Release Notes.html">Release Notes</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">Polly Documentation</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
    
      
    
    <li>Data Lake</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="introduction">Introduction</h1>
<p>With Polly you can access public databases which have been curated and stored in the form of data lakes or make data lakes with your own data. These data lakes can be explored and analyzed either through Polly Notebooks or the several Data Lake Applications available.</p>
<p>Polly Discover consists of the following major components:</p>
<ul>
<li>
<p>Data Lake Curation</p>
<p>Data lakes are reservoirs of information that contain multi-omics data, annotations from publicly available databases, publications, etc. Reservoirs are further segregated into two parts: public data repositories which are curated by Elucidata using public data sources, and private data repositories, where you can add your proprietary data.<br />
<br /></p>
</li>
<li>
<p>Data Lake Exploration </p>
<p>To explore a data lake, Polly provides tools such as applications and Polly Notebooks. These tools enable you to find relevant data by searching for keywords associated with the file name, file metadata, or the contents of a file.<br />
<br /></p>
</li>
<li>
<p>On-the-fly Analysis </p>
<p>Once you have narrowed down relevant omics datasets, you can analyze the dataset(s) on the fly using various statistical analyses while displaying intuitive visualizations.</p>
</li>
</ul>
<h2 id="available-public-data-repositories">Available public data repositories</h2>
<p>Public data repositories on Polly consist of processed and curated datasets from various sources. They can be readily used for searching for new datasets or running an analysis on one or more datasets.</p>
<ul>
<li>
<p>AML: Microarray and RNA Sequencing datasets for Acute Myeloid Leukemia.</p>
</li>
<li>
<p>GBM: Microarray and RNA Sequencing datasets for Gliblastoma Multiforme.</p>
</li>
<li>
<p>IBD: Microarray and RNA Sequencing datasets for Inflammatory Bowel Disease.</p>
</li>
<li>
<p>GEO: Microarray and RNA Sequencing datasets from Gene Expression Omnbius.</p>
</li>
<li>
<p>Single cell Atlas: Single cell RNA Sequencing datasets from Gene Expression Omnibus</p>
</li>
<li>
<p>GTEX: Normal tissue RNA Sequencing datasets from Genotype-Tissue Expression project</p>
</li>
<li>
<p>TCGA: Tumor RNA Sequencing datasets from The Cancer Genome Atlas.</p>
</li>
<li>
<p>COVID-19: Transcriptional datasets for SARS viruses, viral infections, and therapeutics for novel coronavirus.</p>
</li>
</ul>
<p>Additionally, the public data repositories also consist of publicly available databases that have been curated for annotations. These publicly available databases are currently part of these repositories.</p>
<ul>
<li>
<p>HMDB: Pathway information from Human Metabolome Database.</p>
</li>
<li>
<p>KEGG: Pathway information from Kyoto Encyclopedia of Genes and Genomes.</p>
</li>
<li>
<p>Reactome: Pathway information from Reactome.</p>
</li>
<li>
<p>GWAS: Phenotypic data from Genome-Wide Association Studies Catalogue.</p>
</li>
</ul>
<h1 id="dataset-filtering">Dataset Filtering</h1>
<!--## Kibana Dashboard

After selecting a Data Lake, you’ll be able to see a repository overview dashboard. For bigger data lakes,this dashboard is provided to give you an overview of the repository, for instance, the number of data sets in the repository, distribution of tissues, organisms, etc. among the datasets.

![Repository Dashboard](../img/Discover/1.png)

You can filter the datasets of the data lake either by selecting the fields of interest from the input control panel.

![Dashboard Filtering](../img/Discover/2.png)

You can also add your own query. Just click on *add filter* to choose the addition filter field.

![Filter Query](../img/Discover/3.png)

Once these filters are applied, the dashboard will get updated accordingly. -->

<h2 id="dataset-filtering-dashboard">Dataset Filtering Dashboard</h2>
<p>For smaller data lakes, we provide a dataset filtering interface. It allows you to explore and filter the relevant datasets present in the data lake. </p>
<h3 id="filters-and-columns">Filters and Columns</h3>
<p><img alt="Filters and columns" src="../img/Discover/ORC1.png" /></p>
<p>The filtering interface provides <strong>4 parameters</strong> that you can use to filter the datasets within the selected repository. The parameters are:</p>
<ul>
<li>
<p>Disease: This option will give you an overview of all the diseased type datasets present in the repository. You can choose to work on any of the disease options listed or the normal datasets. In order to do the selection, mark the checkboxes present besides the disease of your interest.</p>
</li>
<li>
<p>Organism: It provides the list of the organisms associated with the datasets of the datalake. You can mark a selection to filter the datasets of only the desired organism.</p>
</li>
<li>
<p>Tissue: This section will give you the distribution of tissue across the repository. Click on Load More to look at the entire list, or use the search option to find the tissue type you are looking for. Select the tissue type required to filter the datasets specific to it.</p>
</li>
<li>
<p>Data Type: The dataset variety would be listed in this option. Choose the data type for your study by selecting the checkbox beside it.</p>
</li>
</ul>
<p>When the selections are marked, you can find the filtered datasets on the right panel. </p>
<p><strong>Note:</strong></p>
<ul>
<li>
<p>You can select multiple entries at the same time.</p>
</li>
<li>
<p>To clear your filters at any point in time, click on the clear option present beside all the parameters.</p>
</li>
</ul>
<h3 id="dataset-selection">Dataset Selection</h3>
<p><img alt="Dataset selection" src="../img/Discover/ORC2.png" /></p>
<p>The right panel displays the dataset present in the repository. It incorporates the</p>
<ul>
<li>
<p>Dataset ID: Unique identifier associated with the dataset.</p>
</li>
<li>
<p>Description: It encompasses the title of the paper.</p>
</li>
<li>
<p>Organism: Organism associated with the dataset</p>
</li>
<li>
<p>Datatype: Datatype of the dataset e.g. Transcriptomic, Metabolomics, Single Cell etc</p>
</li>
<li>
<p>Disease: Disease studied with the selected dataset</p>
</li>
<li>
<p>Tissue: Type of tissue the dataset is from</p>
</li>
<li>
<p>Source: Provides the link to the publication</p>
</li>
</ul>
<p>Once you have narrowed down relevant omics datasets, you can mark a selection on the checkbox present beside the desired dataset.</p>
<h3 id="on-the-fly-analysis">On-the-fly Analysis</h3>
<p>You can analyze the selected dataset on the fly using various applications on Polly. They enable you to perform various statistical analyses, displaying intuitive visualizations, and allowing you to create a hitlist while analyzing multiple datasets simultaneously.</p>
<p>In order to select the tool of your analysis, click on <em>Select the Application</em> option at the bottom of the screen after selection of any dataset from the list and choose the analysis platform of your interest and click on open.</p>
<p><img alt="Selecting an application" src="../img/Discover/ORC3.png" /></p>
<p>Select the workspace where you would like to store the analysis and click on <em>Launch</em> to open the selected application/notebook.</p>
<p><img alt="Launching workspace" src="../img/Discover/ORC4.png" /></p>
<h1 id="data-lake-applications">Data Lake Applications</h1>
<p>Data Lake Applications are built on top of data lakes to query and explore relevant datasets. The following data lake applications are a part of the current platform:</p>
<ul>
<li>
<p><strong><a href="https://docs.elucidata.io/Data%20Lake.html#discover">Polly Discover Application:</a></strong></p>
<p>It is a platform for visualization, analytics, and exploration for bulk transcriptomics data curated from GEO. It offers users an interactive dashboard for analysis and visualization of transcriptomics data. Currently, the platform handles gene expression microarray and RNA-seq data and supports three species human, mouse, and rat.<br />
<br /></p>
</li>
<li>
<p><strong><a href="https://docs.elucidata.io/Data%20Lake.html#single-cell-visualization">Single Cell Visualziation:</a></strong></p>
<p>It is a comprehensive visualization platform for single-cell transcriptomics data. The app is helpful in visualizing cells and the association of different genes with the metadata.<br />
<br /></p>
</li>
<li>
<p><strong><a href="https://chanzuckerberg.github.io/cellxgene/posts/annotations">Cellxgene:</a></strong></p>
<p>It is an interactive data explorer for single-cell transcriptomics datasets.<br />
<br /></p>
</li>
<li>
<p><strong>DepMap CCLE:</strong></p>
<p>Exploration application for cell line dependency and gene expression data from DepMap and CCLE.<br />
<br /></p>
</li>
<li>
<p><strong><a href="https://docs.elucidata.io/Data%20Lake.html#gtex">GTEx Application</a></strong></p>
<p>GTEx Application is is a platform for visualization, analytics, and exploration of transcriptomics data from GTEx.<br />
<br /></p>
</li>
<li>
<p><strong><a href="https://docs.elucidata.io/Apps/Metabolomic%20Data/Dual%20Mode%20Visualization.html">Dual Mode Data Visualization(Metabolomics App:</a></strong></p>
<p>This app allows you to perform downstream analysis on untargeted unlabeled metabolomics data along with insightful visualizations. It provides a variety of normalization methods, scaling options, and data visualization functionalities, thereby allowing an efficient analysis of the data to get actionable insights.<br />
<br /></p>
</li>
<li>
<p><strong><a href="https://docs.elucidata.io/Scaling%20compute/Polly%20Notebooks.html">Discover Notebook:</a></strong></p>
<p>This app allows you to perform downstream analysis on untargeted unlabeled metabolomics data along with insightful visualizations. It provides a variety of normalization methods, scaling options, and data visualization functionalities, thereby allowing an efficient analysis of the data to get actionable insights.<br />
<br /></p>
</li>
</ul>
<table>
<thead>
<tr>
<th>Polly Notebooks</th>
<th>Docker</th>
<th>Machine Configuration</th>
</tr>
</thead>
<tbody>
<tr>
<td>Discover Notebook Single-cell</td>
<td>Single Cell Downstream</td>
<td>Memory-optimized 32GB, Polly 2x-large</td>
</tr>
<tr>
<td>Discover Notebook Transcriptomics</td>
<td>RNA-Seq Downstream</td>
<td>RNA-Seq Downstream</td>
</tr>
<tr>
<td>Discover Notebook Proteomics</td>
<td>RNA-Seq Downstream</td>
<td>Polly medium 4GB</td>
</tr>
<tr>
<td>Discover Notebook Metabolomics</td>
<td>Metabolomics</td>
<td>Polly medium 4GB</td>
</tr>
</tbody>
</table>
<h1 id="polly-discover-application">Polly Discover Application</h1>
<p><strong>Opening the app</strong></p>
<p>Upon opening the Discover portal on Polly, choose a data repository that you would like to explore. The page should look something like this.</p>
<p><img alt="Polly Discover" src="../img/Discover/Discover.png" /></p>
<p>After selecting a repository, you’ll be able to view a filtering interface which provides <strong>parameters</strong> that you can use to filter the datasets within the selected repository. Once you select a dataset , you can access the integrated tools attached with the repository. For transcriptomics data, you can use discover application for further analysis of transcriptomics data.</p>
<!-- ![Discover App](../img/Discover/image3.png) -->
<p align="center">
 <img src="../img/Discover/image3.png" alt="Discover App Icon"
    title="Discover App Icon" /> 
</p>

<p>The app shows overview page which contains a brief <strong>description</strong> of the application, it's <strong>scope</strong> and the <strong>usage</strong> as shown below.</p>
<p><img alt="App Description" src="../img/Discover/image4.png" /></p>
<p><strong>Exploring the data lake</strong></p>
<p>Search for relevant datasets by navigating to the <em>Dataset Search</em> tab in the navigation pane to the left. Keyword search can be applied to the following fields:</p>
<ul>
<li>
<p>Data Set ID</p>
</li>
<li>
<p>Data Set Source</p>
</li>
<li>
<p>Description</p>
</li>
<li>
<p>Diseases</p>
</li>
<li>
<p>Is Public</p>
</li>
<li>
<p>Organisms</p>
</li>
<li>
<p>Platform</p>
</li>
<li>
<p>Tissue</p>
</li>
<li>
<p>Year</p>
</li>
</ul>
<p><img alt="Search options" src="../img/Discover/image5.png" /></p>
<p>The search will return all datasets that are associated with your search. The result should look like the image below.</p>
<p><img alt="Search results" src="../img/Discover/image6.png" /></p>
<p>The table shown above shows very few columns by default. In order to view the other columns in the table, you can select the fields from <strong>Available Columns</strong> and click on <em>Show!</em> button. <em>Download Selected Dataset</em> button will let you download the dataset that you have selected on your local system. <em>Export results to CSV</em> button will let you download the search result table in the form of a .csv file. Once you have narrowed down the relevant datasets, you can analyze one or more datasets on the fly within the app.</p>
<p><strong>Analyzing a single dataset</strong></p>
<p>You can analyze a single dataset by selecting the checkbox to the left of the entry in the table. Once you’ve selected the checkbox, click on the <em>Analyze Data</em> button below the table description.</p>
<p><img alt="Select a data set" src="../img/Discover/image7.png" /></p>
<p>After clicking the <em>Analyze Data</em> button, the app will read the selected dataset and take you to the <em>Dataset Analysis</em> tab. Here, you can perform the following analyses:</p>
<ul>
<li>
<p><strong>Principal Component Analysis (PCA)</strong></p>
<p>Principal Component Analysis: Also known as PCA plot, it is used to see the overall differences between cohorts of interest, if a strong separation is found along X axis (PC1) then that means strong biological differences between cohorts of interest. One can also increase the number of genes considered in the PCA plot, as one increases the number of genes, it is bound to decrease the PC1 component.<br />
<br /></p>
</li>
<li>
<p><strong>Boxplot Visualization</strong></p>
<p>Boxplot can be really useful in understanding the distribution of expression within a dataset. For any downstream analysis such as differential expression or pathway analysis, the distribution has to be normal since they use tests which assume this distribution.<br />
<br /></p>
</li>
<li>
<p><strong>Plots</strong></p>
<p>A box and whisker plot (a boxplot) is a graph that presents information from a five-number summary namely lower extreme, lower quartile, median, upper quartile, and upper extreme. In this plot, the median is marked by a vertical line inside the box; the ends of the box are upper and lower quartiles; the two lines outside the box extend to the highest and lowest observations. It is useful for knowing the nature of distribution (i.e., skewed) and potential unusual observations.<br />
<br /></p>
</li>
<li>
<p><strong>Heatmap</strong></p>
<p>A heatmap is a graphical representation of data that uses a system of color-coding to represent different values. This heatmap shows the cohort wise mean expression of a particular gene. The samples are aggregated on the basis of a given cohort and the mean is calculated based on the cohort information.<br />
<br /></p>
</li>
<li>
<p><strong>Differential Expression</strong></p>
<p>Differential expression analysis means taking the normalised read count data and performing statistical analysis to discover quantitative changes in expression levels between experimental groups. For example, we use statistical testing to decide whether, for a given gene, an observed difference in read counts is significant, that is, whether it is greater than what would be expected just due to natural random variation.<br />
<br /></p>
</li>
<li>
<p><strong>X2K Analysis</strong></p>
<p>X2K infers upstream regulatory networks from signatures of differentially expressed genes. By combining transcription factor enrichment analysis, protein-protein interaction network expansion, with kinase enrichment analysis, X2K produces inferred networks of transcription factors, proteins, and kinases predicted to regulate the expression of the inputted gene list.<br />
<br /></p>
</li>
<li>
<p><strong>Gene Ontology Plot</strong></p>
<p>Gene Ontology Annotation Plot is a simple but useful tool for visualizing, comparing and plotting GO (Gene Ontology) annotation results.<br />
<br /></p>
</li>
<li>
<p><strong>Enrichr</strong></p>
<p>Enrichr, includes new gene-set libraries, an alternative approach to rank enriched terms, and various interactive visualization approaches to display enrichment results using the JavaScript library, Data Driven Documents (D3).<br />
<br /></p>
</li>
<li>
<p><strong>GSEA</strong> </p>
<p>Gene Set Enrichment Analysis (GSEA) is a computational method that determines whether an a priori defined set of genes shows statistically significant, concordant differences between two biological states (e.g. phenotypes).<br />
<br /></p>
</li>
<li>
<p><strong>Specific Pathway Visualization using Pathview</strong></p>
<p>Pathview maps, integrates and renders a wide variety of biological data on relevant pathway graphs.<br />
<br /></p>
</li>
</ul>
<p><img alt="Analyses possible" src="../img/Discover/image8.png" /></p>
<!--
**Analyzing multiple datasets**

It is also possible to analyze multiple datasets simultaneously. If you select multiple datasets from the search result and click the *Analyze Data* button, you will see a new *Multiple Analyses* Tab, wherein you can select algorithms that you want to run on the selected datasets. While selecting algorithms, a workflow of nodes is formed. These nodes are input blocks for each algorithm for each dataset you selected.

**Total number of nodes = number of datasets selected (n) X number of algorithms selected (m)**

![Multiple analyses](../img/Discover/image9.png)

Upon clicking each node, a pop up with input parameters for each algorithm corresponding to a particular dataset will appear. Adjust these parameters to fit your analysis before executing the algorithms.

![Algorithm input parameters](../img/Discover/image10.png)

Once you have filled all the parameters, you can verify them in the table formed below. Next, execute the algorithms by clicking on the *Execute Algorithms* button.

![Execute algorithms](../img/Discover/image11.png)

After the algorithms have been executed, you’ll be taken to the rules tab, where you can define the rules by which you want to filter the output of the algorithms to get a preliminary hitlist. For instance, after running differential expression, you want to filter by logFC and *p*-value. Along with that you can also add annotations to that hitlist by selecting your source of annotations.

![Rules tab](../img/Discover/image12.png)

Finally, on clicking the *Execute Rules* button, you will be able to get a preliminary hitlist.

![Preliminary hitlist](../img/Discover/image13.png)


##DepMap CCLE

**Opening the app**

Upon opening the *Discover Insights* module on Polly, choose the *Depmap Data* repository.

![Selecting app](../img/Discover/image31.png)

After selecting the *Depmap repository*, you’ll be able to see a dashboard and the *DepMap CCLE application* icon. The dashboard will give you an overview of the repository, for instance, the information of different cell lines in the  repository, distribution of tissue lineages, metastatic status of samples, etc. among the datasets.

![Kibana dashboard](../img/Discover/image32.png)

Click on the below icon to start the *DepMap CCLE application*.

![DepMap icon](../img/Discover/image33.png)

Once inside the application, you can find two examples on the home page that briefly describes the usage of the application.

![App overview](../img/Discover/image34.png)

**Exploring the data lake**

This application provides the functionality to explore the DepMap CCLE data lake. You can query the data lake for three different functionalities for different genes in the application. Each functionality has a separate tab on the left for easy access. 

*   Gene Essentiality

*   Gene Cluster

*   CCLE Heatmap

**Gene Essentiality**

Gene Essentiality of the gene describes the efficacy vs selectivity map for a gene based on different models. The terms can be defined as: 

*   **Efficacy**: how essential the gene is in the sensitive cell lines (the more negative the efficacy is, the more essential the gene is).

*   **Selectivity**: how selectively essential the gene is between sensitive and resistant cell lines (the more positive the selectivity is, the more selective the gene is).

As shown in the following image, you can search for a gene and its essentiality. Click on *Press Go* to check the scores. 

![Select gene](../img/Discover/image35.png)

The search returns an *essentiality map* on the left panel with the searched gene highlighted in red color. The right panel describes the efficacy and selectivity scores of the *matched gene* and the *dependency scores* of different models in the data lake. 

![Essentiality](../img/Discover/image36.png)

![Matched genes](../img/Discover/image37.png)

![Dep score](../img/Discover/image38.png)

**Gene Cluster**

In concept, genes that work together as complexes or pathways should show similar dependency scores across cell lines. With this assumption, 2,492 essential genes were clustered based on the similarity of their dependency scores across 423 cell lines; the resulting cluster groups essential genes that likely work together.

As shown in the figure below, you can query for your gene of interest to see which genes cluster with it. To search, type in the name of the gene. Suggestions will be displayed to help you select the gene of your interest. The *cluster size* option changes the parameters for stringency of clustering. A small cluster equates to stringent cutoffs.  

![Cluster gene select](../img/Discover/image39.png)

The query returns an *efficacy vs selectivity plot* of the gene, a *t-SNE Plot* for functional similarity and the *clustered genes* with the searched gene in the tabular format along with *connectivity map* and *correlation scores*. 

![Eff plot](../img/Discover/image40.png)

![tSNE](../img/Discover/image41.png)

![Clustered genes](../img/Discover/image42.png)

**CCLE Heatmap**

*CCLE heatmap* aids you in visualizing the cancer specific expression of selected genes in different cell lines models along with the dependency scores of different genes. The heatmap requires three inputs, *Cancer Type* to select the cell lines models, *Dependency Genes* for dependency scores of genes of interest and *Genes for Expression* to display the expression of the genes. The inputs can be provided as shown in the following image. 

![Select cancer](../img/Discover/image43.png)

Clicking on *Plot* processes the data and the result of the query is a heatmap with columns as the sample, rows for the genes and column descriptors describing the dependency scores of the genes. 

![Heatmap](../img/Discover/image44.png)

##Single Cell Visualization

Single Cell Visualization provides a visualization interface for single cell data hosted on GEO. The data lake working on its backend is updated regularly with new datasets. The app has different visualizations for understanding data: *Dimensionality reduction plot*, *Violin plot*, *Dot Plot*. 
-->

<h2 id="gtex">GTEx</h2>
<p><strong>Opening the app</strong></p>
<p>GTEx respository can be accessed using the GTEx card on Discover. </p>
<p><img alt="Polly GTEx" src="../img/Discover/GTEx.png" /></p>
<p>After selecting the repository, you’ll be able to see a dashboard with different tissues. Select a dataset and use GTEx application to explore the dataset. </p>
<p><img alt="Repository Dashboard" src="../img/Discover/GTEx_2.png" /></p>
<p>The app will open and you should see the overview page which contains a brief overview of the application, scope and caveats as shown below.</p>
<p><img alt="App Description" src="../img/Discover/GTEx_3.png" /></p>
<p><strong>Analyzing a dataset</strong></p>
<p>As the application starts, it will load the requested dataset. Once it is loaded, it can be explored. </p>
<ul>
<li><strong>Principal Component Analysis (PCA)</strong></li>
</ul>
<p>Principal Component Analysis: Also known as PCA plot, it is used to see the overall differences between cohorts of interest, if a strong separation is found along X axis (PC1) then that means strong biological differences between cohorts of interest. It provides an aspect to check the quality control of different samples. </p>
<p><img alt="Metadata Table" src="../img/Discover/GTEx_4.png" /></p>
<p>The tab provides a metadata table to check different characterstics of samples. Furthermore various parameters of PCA can be adjusted. </p>
<p><img alt="PCA Parameters" src="../img/Discover/GTEx_5.png" /></p>
<p>A publication quality and an interactive version of the PCA plot is available to explore. </p>
<p><img alt="PCA Plot" src="../img/Discover/GTEx_6.png" /></p>
<ul>
<li><strong>Bar Plot</strong></li>
</ul>
<p><em>Barplot</em> provides exploration of different genes either standalone or as a part of different pathways. The distribution can be grouped by different metadata cohorts such as tissue type or tissue-subtype. </p>
<p>Using the <em>Gene Expression</em> toggle, different genes can be queried for different samples. </p>
<p><img alt="Gene Expression" src="../img/Discover/GTEx_7.png" /></p>
<p>Upon selecting the <em>Pathway Visualization</em> option, pathway specific genes can be selected. At a given time more than one pathways can be selected  </p>
<p><img alt="Pathway Visualization" src="../img/Discover/GTEx_8.png" />
<img alt="Pathway Visualization Plot" src="../img/Discover/GTEx_9.png" /></p>
<ul>
<li><strong>GTEx Expression Map</strong></li>
</ul>
<p><em>GTEx Expression Map</em> can be used to explore the distribution of selected genes in different GTEx tissues. After exploring the selected tissue and finding list of genes of interest, it presents visualization methods like <em>GTEx Expression Violin</em> and <em>GTEx Expression Heatmap</em> to study the distribution of genes across different tissues. </p>
<p>A single gene can be selected to plot violin for it's expression across different tissues. </p>
<p><img alt="GTEX Expression Violin" src="../img/Discover/GTEx_10.png" /></p>
<p>Multipe genes can be used to make a heatmap for different tissues. </p>
<p><img alt="GTEx Expression Heatmap" src="../img/Discover/GTEx_11.png" /></p>
<h2 id="single-cell-visualization">Single Cell Visualization</h2>
<p><strong>Opening the app</strong></p>
<p>Upon opening the Discover application on Polly, choose a relevant data repository which hosts single cell data. </p>
<p><img alt="Polly Discover" src="../img/Discover/Discover.png" /></p>
<p>After selecting a repository, you’ll be able to view a filtering interface which provides <strong>parameters</strong> that you can use to filter the datasets within the selected repository. Once you select a dataset you can access the integrated tools attached with the repository. You can use Single Cell Visualization application for further analysis of single cell data.</p>
<p><img alt="single cell App" src="../img/Discover/single_2.png" /></p>
<p>The app will open and you should see the overview page which contains a brief overview of the application, scope and caveats as shown below.</p>
<p><img alt="App Description" src="../img/Discover/single_3.png" /></p>
<p><strong>Exploring the data lake</strong></p>
<p>Search for relevant datasets by navigating to the <em>Dataset Search</em> tab in the navigation panel to the left. Keyword search can be applied to the following fields:</p>
<ul>
<li>
<p>DatasetID</p>
</li>
<li>
<p>Platform</p>
</li>
<li>
<p>Title</p>
</li>
<li>
<p>Description</p>
</li>
<li>
<p>Disease</p>
</li>
<li>
<p>Pubmed ID</p>
</li>
<li>
<p>Organism</p>
</li>
<li>
<p>Cell Types</p>
</li>
<li>
<p>Tissue</p>
</li>
</ul>
<p><img alt="Search options" src="../img/Discover/single_4.png" /></p>
<p>The search will return all datasets that are associated with your search. The result should look like as shown below.</p>
<p><img alt="Search results" src="../img/Discover/single_5.png" /></p>
<p>The table shown above shows very few columns by default. In order to view the other columns in the table, you can select the fields from <strong>Available Columns</strong> and click on <em>Show</em> button. Once you have narrowed down the relevant datasets, you can analyze one dataset on the fly within the app.</p>
<p><strong>Analyzing a dataset</strong></p>
<p>You can analyze a single dataset by selecting the checkbox to the left of the entry in the table. Once you’ve selected the checkbox, click on the <em>Load</em> button below the table description.</p>
<p><img alt="Select a data set" src="../img/Discover/single_6.png" /></p>
<p>After clicking the <em>Load</em> button, the app will read the selected dataset. Once the loading finishes, you can check the further tabs to explore the dataset:</p>
<ul>
<li><strong>Dataset Summary</strong></li>
</ul>
<p>This tab provides quick summary of the selected dataset. The tab reveals the no. of celltype/clusters, genes and cells, available metadata and quality control metrics for the selected dataset. The Value boxes at the top provide information about the no. of genes, cells and celltypes/clusters. </p>
<p><img alt="Value Boxes" src="../img/Discover/single_7.png" /></p>
<p>Below it lies the metadata summary table which contains the different metadata fields and their categories. The table is searchable and clicking on a particular metadata shows it's distribution. For instance if you want to see the distribution of cell types in a study, you can search the keyword 'cell_type' in <em>name</em> search box. Upon click on it a table describing the distribution of cell types will popup.</p>
<p><img alt="Metadata Table" src="../img/Discover/single_8.png" /></p>
<p>Quality Control(QC) Metrics helps in understanding the processing of the dataset. The application provides the opportunity to understand quality control using <em>QC distribution</em> and <em>QC scatter</em> plots. Using QC distribution one can understand the distribution of a single quality control metric in a particular metadata. For instance, if we want to check the distribution of gene counts in different cell type, we can select 'gene_counts' as QC metric and 'cell_type' as cluster.</p>
<p><img alt="QC Distribution" src="../img/Discover/single_9.png" /></p>
<p>Using QC scatter plot one can understand the assosciation between distribution of two different quality control metrics. It can be useful in understanding the distribution of gene counts and UMI counts.</p>
<p><img alt="QC Scatter" src="../img/Discover/single_10.png" /></p>
<ul>
<li><strong>Cell Visualization</strong></li>
</ul>
<p><em>Cell visualization</em> provides exploration of cells using dimensionality reduction methods. The tab presents the dimensionality reduction methods such as tSNE, UMAP, PCA and others to visualize the distribution of the cells. The <em>Visualize cells</em> panel on the right, shows the distribution of the cells in an interactive way. </p>
<p><img alt="Visualize cells" src="../img/Discover/single_11.png" /></p>
<p>Using <em>Feature selection</em> panel, a metadata or a gene feature can be selected for plotting. </p>
<p><img alt="Feature selection" src="../img/Discover/single_12.png" /></p>
<p>The <em>Customize visualization</em> panel, offers the scope of customizing visualization features such as <em>highlight non-zero cells</em>, <em>point size</em> and <em>method</em> used based on personal preferences.</p>
<p><img alt="Customize visualization" src="../img/Discover/single_13.png" />       </p>
<ul>
<li><strong>Marker</strong></li>
</ul>
<p><em>Marker</em> tab provides exploration of distribution of markers. It presents visualization methods like <em>dot plot</em> and <em>violin plot</em> to study the distribution of genes across different metadata. The <em>Marker selection</em> panel, provides the user with options to choose different genes and metadata for plotting marker distribution. </p>
<p><img alt="Marker selection" src="../img/Discover/single_14.png" /></p>
<p>The <em>Marker dot plot</em> panel, is the area for exploring the average expression and distribution of marker using dot plot in an interactive fashion.</p>
<p><img alt="Marker Dot plot" src="../img/Discover/single_15.png" /></p>
<p>The <em>Marker Violin plot</em> panel is the area for exploring the marker distribution using violin plot in an interactive fashion. Using <em>Customize Violin</em> slider, the range of values used for plotting violin can be adjusted. It is useful for observing a section of data such as the non-zero values of the expression. </p>
<p><img alt="Marker Violin plot" src="../img/Discover/single_16.png" /></p>
<ul>
<li><strong>Cell-Type Aggregation</strong></li>
</ul>
<p>After having insights about different markers in a single cell study, it is imperative to have a look at the expression of the selected markers across different cell types in different studies. Single cell visualization provides pan-dataset exloratory analysis in the form of <em>Cell Type Aggregation</em>. It provides the scope to query the respective repository for median expression of a selected gene across top 20 cell types based on expression. This provides the scope to study expression of a gene in a particular cell type in a given biological context. </p>
<p><img alt="cell-type aggregation" src="../img/Discover/single_17.png" /></p>
<p>Potential use-case example: What is the expression of SARS-CoV2 virus entry specific host protein 'ACE2' across different cell types in different studies? </p>
<p><img alt="Use case" src="../img/Discover/single_18.png" /></p>
<p>The <em>Cell Type Aggregation</em> tab provides the input in the form of text field where query can be made for a gene, using our internal discover services, the input gene is queried across all the datasets in which it is expressed in the repository. To check for ACE2 expression, simply enter ACE2 in the search field. After entering the gene, clicking on <em>Search gene</em> will generate a bar plot showing the median expression and the distribution of gene expression across different studies.</p>
<p><img alt="Cell-type aggregation plot" src="../img/Discover/single_19.png" /></p>
<h1 id="access-through-polly-notebook-interface">Access through Polly Notebook interface</h1>
<p>The Polly Notebook Dockers on Polly have an <strong>internal</strong> python package called ‘discoverPy’ pre-installed, which can be used to search for datasets in the various data repositories.</p>
<h2 id="structure-of-a-data-repository">Structure of a data repository</h2>
<p>A data repository is a collection of different files having different file types. To ensure easy access at a granular level to all datasets a data repository is organized in the following manner. Under this schema, each repository can be considered as a collection of indices which can be used for querying. The discoverPy package can access all indices of a data repository using API endpoints.</p>
<p><img alt="Structure of a data repository" src="../img/Discover/discover_data_repository.png" /></p>
<p>Click <a href="https://docs.elucidata.io/Scaling%20compute/Polly%20Notebooks.html">here</a> for a detailed documentation about Polly Notebooks.</p>
<h2 id="usage">Usage</h2>
<ul>
<li>Initialize a discover object</li>
</ul>
<p>This discover object is used to interact with a dataset repository.</p>
<pre><code>from discoverpy import Discover
discover = Discover() 
discover</code></pre>

<p><img alt="Discover object" src="../img/Discover/image14.png" /></p>
<ul>
<li>List all available data repositories along with their indices.</li>
</ul>
<pre><code>discover.get_repositories()</code></pre>

<p><img alt="Endpoints for public data" src="../img/Discover/image15.png" /></p>
<ul>
<li>
<p>Set a repository for fetching the different endpoints.</p>
<p>Choose a repository from the list of repositories and use it's corresponding it to set the discover object to point to that repository.</p>
<ul>
<li>
<p>For single cell repositories use mode='single_cell'.</p>
</li>
<li>
<p>For bulk data repositories use mode='bulk' (default)</p>
</li>
</ul>
</li>
</ul>
<p>For <em>geo</em> repository <em>repo_id</em> is <strong>16</strong>.</p>
<pre><code>discover.set_repo('16')</code></pre>

<p>For <em>sc_data_lake</em> repository <em>repo_id</em> is <strong>17</strong>.</p>
<pre><code>discover.set_repo("17", mode="single_cell")</code></pre>

<p>After you’ve added the indices for a repository, you can view the discover object</p>
<pre><code>discover</code></pre>

<p><img alt="View the discover object" src="../img/Discover/image16.png" /></p>
<p>Note that the ‘annotation_repo’ index is added automatically for each repository.</p>
<h2 id="querying-at-the-dataset-level">Querying at the dataset level</h2>
<p>To search for datasets, the ‘_files’ index can be searched using the metadata fields present in it.</p>
<ul>
<li>Get fields present in the index</li>
</ul>
<pre><code>discover.dataset_repo.get_all_fields()</code></pre>

<p><img alt="Fields present in the index" src="../img/Discover/image17.png" /></p>
<ul>
<li>
<p>To get a sense of what values are present in each field, one can view the top n entries. Some generic fields are present for each file.</p>
<ul>
<li>
<p>__bucket__: S3 bucket name</p>
</li>
<li>
<p>__filetype__: Type of file such as pdf, gct etc.</p>
</li>
<li>
<p>__key__: S3 key of the file</p>
</li>
<li>
<p>__location__: Location of the file within data repository</p>
</li>
</ul>
</li>
</ul>
<pre><code>discover.dataset_repo.get_top_n_examples(n = 30)</code></pre>

<p><img alt="Values present in each field" src="../img/Discover/image18.png" /></p>
<ul>
<li>Search for a dataset by keyword in a particular field. Searching for “mll” in the field “description” here.</li>
</ul>
<pre><code>dataset_query_df = discover.dataset_repo.query_dataset_by_field("description","mll") dataset_query_df</code></pre>

<p><img alt="Search for a dataser" src="../img/Discover/image19.png" /></p>
<h2 id="querying-at-the-sample-level">Querying at the sample level</h2>
<p><strong>GCT File Format</strong></p>
<p>The datasets in the public repositories are saved as a .gct file. This is a file format in which data can be stored along with the sample metadata. The data values in the actual matrix along with features (genes) are indexed in the ‘_gct_data’ index of the repository and the sample metadata is index in the ‘_gct_metadata’ of the index of the repository.</p>
<p><img alt="GCT file structure" src="../img/Discover/gct_file.png" /></p>
<p><strong>H5AD File Format</strong></p>
<p>The single cell datasets in the public repositories are saved as a .h5ad file. This is a file format in which data can be stored along with the sample metadata.</p>
<p><img alt="H5AD file structure" src="../img/Discover/h5ad_file.svg" /></p>
<ul>
<li>Get fields present in the index</li>
</ul>
<pre><code>discover.sample_repo.get_all_fields()</code></pre>

<p><img alt="Fields present in the index" src="../img/Discover/image20.png" /></p>
<ul>
<li>To get a sense of what values are present in each field, one can view the top n entries.</li>
</ul>
<pre><code>discover.sample_repo.get_top_n_examples(n =30)</code></pre>

<p><img alt="Values present in field" src="../img/Discover/image21.png" /></p>
<ul>
<li>Search for samples by keyword in a particular field. Searching for “M1” in the field “fab_classification_ch1” here.</li>
</ul>
<pre><code>fab_df = discover.sample_repo.query_samples_by_field("fab_classification_ch1", "M1", n = 100)
fab_df</code></pre>

<p><img alt="Search for samples using keywords" src="../img/Discover/image22.png" /></p>
<ul>
<li>Search for samples by keywords in all fields. This can be used if the field to search for is not known beforehand.</li>
</ul>
<pre><code>fab_all_fields = discover.sample_repo.query_samples_by_all_fields("M1", n = 100) fab_all_fields</code></pre>

<p><img alt="Search for samples using keywords" src="../img/Discover/image23.png" /></p>
<h2 id="querying-at-the-feature-level">Querying at the feature level</h2>
<p>The matrix of a .gct/.h5ad file contains the actual values for the different features(genes/metabolites). The ‘_gct_index’ or ‘_h5ad_index’ index of a repository can be queried for features.</p>
<ul>
<li>Get fields present in the index  </li>
</ul>
<pre><code>discover.feature_repo.get_all_fields()</code></pre>

<p><img alt="Fields present in the index" src="../img/Discover/image24.png" /></p>
<ul>
<li>To get a sense of what values are present in each field, one can view the top n entries.</li>
</ul>
<pre><code>discover.feature_repo.get_top_n_examples(n=20)</code></pre>

<p><img alt="Values present in field" src="../img/Discover/image25.png" /></p>
<p>The ‘__index__’ column contains the feature name</p>
<ul>
<li>Get values for a particular feature across all samples in all datasets of the repository. Getting values for “NRAS” gene here.</li>
</ul>
<pre><code>nras_df =discover.feature_repo.get_feature_values("NRAS", n = 1000)
nras_df</code></pre>

<p><img alt="Values for a feature" src="../img/Discover/image26.png" /></p>
<ul>
<li>To get features from all single cell datasets, use the variant get_feature_values_sc. See the following example.</li>
</ul>
<pre><code>hhex_df =discover.feature_repo.get_feature_values_sc("HHEX", n = 1000) 
hhex_df</code></pre>

<p><img alt="Values for a feature in single cell datasets" src="../img/Discover/image260.png" /></p>
<h2 id="access-annotation-repositories">Access annotation repositories</h2>
<p>The various gene annotation databases can also be accessed through discoverpy. These can be used to get information about a particular gene or a set of genes.</p>
<ul>
<li>Get all annotation databases</li>
</ul>
<pre><code>discover.annotation_repo.get_annotation_databases()</code></pre>

<p><img alt="Get all annotation databases" src="../img/Discover/image27.png" /></p>
<ul>
<li>Get annotations for a list of genes from a particular database. Getting Reactome pathways for the genes.</li>
</ul>
<pre><code>discover.annotation_repo.get_feature_annotation('reactome', ['ACTA1', 'AHCTF1', 'AKAP13', 'ATP2C1', 'CDK7'])</code></pre>

<p><img alt="Get annotations for a list of genes" src="../img/Discover/image28.png" /></p>
<h2 id="advanced-queries">Advanced queries</h2>
<p>You can also perform more complex queries on multiple fields combining them with boolean logic. Some examples are shown here.</p>
<ul>
<li>Get microarray stem cell datasets which did not involve a knockdown experiment</li>
</ul>
<pre><code>discover.dataset_repo.query_dataset_by_field_combination(and_fields={"platform":"Microarray", "tissue":"stem cells"}, not_fields={"description":"knockdown"}, n = 50)</code></pre>

<p><img alt="Advanced query example 1" src="../img/Discover/image29.png" /></p>
<ul>
<li>Get samples containing CD34 cells or mononuclear cells do not include de novo samples</li>
</ul>
<pre><code>discover.sample_repo.query_samples_by_field_combination(or_fields = {"cell_type_ch1":"CD34","cell_type_ch1":"mononuclear"}, not_fields = {"treatment_protocol_ch1":"de novo"}, n = 300)</code></pre>

<p><img alt="Advanced query example 2" src="../img/Discover/image30.png" /></p>
<h2 id="downloading-datasets">Downloading datasets</h2>
<ul>
<li>
<p>You can use the get_file(key, repo_id, file_name) function to download a dataset from a datalake repository. The function has following 3 parameters:</p>
<ul>
<li>
<p>__key__: S3 key of the file</p>
</li>
<li>
<p>__repo_id__: Repository id</p>
</li>
<li>
<p>__file_name__: Name of the file with file extentions such as gct, h5ad etc.</p>
</li>
</ul>
</li>
</ul>
<pre><code>discover.get_file('AML_data_lake/data/Microarray/GSE76320/GCT/GSE76320_GPL8321_curated.gct', '1', 'GSE76320_GPL8321_curated.gct')</code></pre>

<h1 id="videos">Videos</h1>
<div style="position: relative; padding-bottom: 56.25%; height: 0; overflow: hidden; max-width: 100%; height: auto;">
    <iframe src="https://www.youtube.com/embed/AYgAb5Lbj4g" frameborder="0" allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;"></iframe>
</div>

<p><br></p>
<div style="position: relative; padding-bottom: 56.25%; height: 0; overflow: hidden; max-width: 100%; height: auto;">
    <iframe src="https://www.youtube.com/embed/J-l298jBtIQ" frameborder="0" allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;"></iframe>
</div>

<p><br></p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Release Notes.html" class="btn btn-neutral float-right" title="Release Notes">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="Scaling compute/Polly Notebooks.html" class="btn btn-neutral" title="Notebooks"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="Scaling compute/Polly Notebooks.html" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="Release Notes.html" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '.';</script>
    <script src="js/theme.js" defer></script>
      <script src="search/main.js" defer></script>

</body>
</html>
