
# What is OmixAtlas?

Every year, vast amounts of biological multi-omics data are being generated and deposited in repositories by academic labs and organizations worldwide. These data hold tremendous potential for reuse and discovery but are scattered across multiple, disparate sources and lack standardization. Thus, the availability of data does not equate to its easy usability. This makes the need for an efficient means of exploring molecular data an immediate necessity. 

Additionally, no single type of data, be it metabolomic, proteomic or genomic, will be sufficient to capture the complexity of biological phenomena. OmixAtlas aims to address these issues by ensuring that the metadata from different data types and across different data sources have been curated and harmonized and made ready for downstream machine learning and analytical applications.

OmixAtlas is a collection of datasets with curated biomolecular data for understanding human physiology and disease pathology. It aims to adopt an integrated approach with access to ML-ready biology-centric data from public and proprietary repositories to generate new insights. 

Take a data-centric approach to research with ML-ready biomolecular data on OmixAtlas.

# What are the key features of OmixAtlas?

1. Multi-omics data in one place: OmixAtlas provides access to over 26 different data types from over 30 data sources.

2. Continuously evolving: In keeping with the rapid pace of biological data generation, the OmixAtlas is updated with a frequency that is in sync with the source repositories, wherever applicable.

3. ML-ready data made available through Curation: All types of data from diverse sources have been structured, metadata have been harmonized with controlled vocabularies and ontologies and made ready for downstream analysis using various applications.

4. Integratability: OmixAtlas is made available on Polly’s data infrastructure, which allows querying of the data through point and click solutions and advanced querying through polly libraries. Code-based advanced access means that you can use these data on Polly or outside of Polly on a platform of your choice.

# How is OmixAtlas different from available biomedical data repositories?
The data in OmixAtlas are curated through Polly’s ML-based curation workflow that structures different types of data, harmonizes metadata, and makes the data analysis ready. All data available in OmixAtlas can be queried and directly used in downstream statistical or ML-based analyses. With OmixAtlas one can:

- Access data from heterogeneous sources: OmixAtlas has over 26 data types from over 30 public repositories and other sources in one central location. You can access study, samples, data files, and associated metadata all in one place.

- Leverage curation infrastructure to standardize data: Build machine learning applications faster with analysis-ready data. Use Polly’s curation models to process public and proprietary data through standard ingestion pipelines.

- Query heterogeneous biomolecular data with code: Perform integrative analysis with powerful code-based querying capabilities across the OmixAtlas data catalog. Use Polly Libraries for exploring data in depth through code.

- Access and analyze data on any computational environment: Stream data to your compute infrastructure from OmixAtlas. Focus on analysis while OmixAtlas takes care of data storage and management.

- Manage in-house data at scale with Enterprise OmixAtlas: Harmonize proprietary data using the scalable curation infrastructure of Polly, making it ML-ready for discovering new insights.

# How can OmixAtlas benefit the user?
|**Working with multi-omics data**|**How our OmixAtlases can help**|
| :-: | :-: |
|Finding multi-omics data relevant to me can take weeks as data is semi-structured and scattered across several sources|Access to thousands of tissue-derived or disease specific multi-omics datasets from multiple sources in one place|
|Enriching my data and preparing it for machine learning applications takes time and effort that could be better spent on insight discovery|OmixAtlas data is processed through a standard pipeline and enriched with harmonized, scientifically relevant metadata, ready for machine learning applications|
|I want access to curated data but don’t want to switch to another computational platform|OmixAtlas data can be accessed and analyzed on your own existing computational infrastructure|

# OmixAtlas Case Studies

1. [Integrative single-cell transcriptome analysis reveals a subpopulation of fibroblasts associated with favorable prognosis of liver cancer patients](`Case Studies`/`Single Cell Integration`/`Single Cell Integration.md`)
2. [Predicting survivability of patients with  Hepatocellular Carcinoma using multi-omics data](`Case Studies`/HCC_survival_prediction-Polly_case_study/HCC_survival_prediction-Polly_case_study.md) 
3. [Automated cell-type annotation in a liver single cell transcriptomics data](`Case Studies`/`Celltype Identification`/Celltype_identification.md)
