<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../../img/elucidata.ico">
  
  <title>Dual Mode Data Visualization - Polly Documentation</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Dual Mode Data Visualization";
    var mkdocs_page_input_path = "Apps/Metabolomic Data/Dual Mode Visualization.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../../js/jquery-2.1.1.min.js" defer></script>
  <script src="../../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href="../../index.html" class="icon icon-home"> Polly Documentation</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="../../index.html">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Getting Started</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../../Getting Started/Workspaces.html">Workspaces</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Apps</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../Introduction.html">Introduction</a>
                </li>
                <li class="">
                    
    <a class="" href="../Host Apps.html">Host Apps</a>
                </li>
                <li class=" current">
                    
    <span class="caption-text">Metabolomic Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="El-MAVEN Phi Relative LC-MS.html">El-MAVEN Phi Relative LC-MS</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="MetScape.html">MetScape</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="FirstView.html">FirstView</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Untargeted Pipeline.html">Untargeted Pipeline</a>
                </li>
                <li class="toctree-l3 current">
                    
    <a class="current" href="Dual Mode Visualization.html">Dual Mode Data Visualization</a>
    <ul class="subnav">
            
    <li class="toctree-l4"><a href="#introduction">Introduction</a></li>
    
        <ul>
        
            <li><a class="toctree-l5" href="#overview">Overview</a></li>
        
            <li><a class="toctree-l5" href="#scope-of-the-app">Scope of the App</a></li>
        
        </ul>
    

    <li class="toctree-l4"><a href="#getting-started">Getting Started</a></li>
    
        <ul>
        
            <li><a class="toctree-l5" href="#user-input">User Input</a></li>
        
        </ul>
    

    <li class="toctree-l4"><a href="#caveats">Caveats</a></li>
    

    <li class="toctree-l4"><a href="#tutorial">Tutorial</a></li>
    
        <ul>
        
            <li><a class="toctree-l5" href="#upload-files">Upload Files</a></li>
        
            <li><a class="toctree-l5" href="#pre-processing">Pre-processing</a></li>
        
            <li><a class="toctree-l5" href="#quality-checks">Quality Checks</a></li>
        
            <li><a class="toctree-l5" href="#statistical-analysis">Statistical Analysis</a></li>
        
            <li><a class="toctree-l5" href="#visualization">Visualization</a></li>
        
            <li><a class="toctree-l5" href="#intomix-input">IntOmix Input</a></li>
        
            <li><a class="toctree-l5" href="#heatmap">Heatmap</a></li>
        
            <li><a class="toctree-l5" href="#comparative-analysis">Comparative Analysis</a></li>
        
            <li><a class="toctree-l5" href="#data-for-metaboanalyst">Data for MetaboAnalyst</a></li>
        
        </ul>
    

    </ul>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="QuantFit.html">QuantFit</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="El-MAVEN.html">El-MAVEN</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Labeled LC-MSMS Workflow.html">Labeled LC-MS/MS Workfow</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="Labeled LC-MS Workflow.html">Labeled LC-MS Workflow</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="CompoundDiscoverer QuantFit.html">Compound Discoverer<sup>TM</sup> QuantFit</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Screening Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../Screening Data/CRISPR Screening.html">CRISPR Screening</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../Screening Data/High Throughput Drug Screening.html">High Throughput Drug Screening</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Sequencing Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../Sequencing Data/RNA Seq Workflow.html">RNA Seq Workflow</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Proteomic Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../Proteomic Data/Proteomics Workflow.html">Proteomics Workflow</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Lipidomic Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../Lipidomics Data/Lipidomics Visualization Dashboard.html">Lipidomics Visualization Dashboard</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Multi-omic Data</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../Multi-omic Data/IntOmix.html">IntOmix</a>
                </li>
    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Data Studio</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../Data Studio/Data Studio.html">Introduction</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../Data Studio/CCLE Correlation Analysis.html">CCLE Correlation Analysis</a>
                </li>
    </ul>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Scaling Compute</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../../Scaling compute/Polly CLI.html">Polly CLI</a>
                </li>
                <li class="">
                    
    <a class="" href="../../Scaling compute/Polly Notebooks.html">Notebooks</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../../Data Lake.html">Data Lake</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../../Release Notes.html">Release Notes</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../../index.html">Polly Documentation</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../../index.html">Docs</a> &raquo;</li>
    
      
        
          <li>Metabolomic Data &raquo;</li>
        
      
        
          <li>Apps &raquo;</li>
        
      
    
    <li>Dual Mode Data Visualization</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="introduction">Introduction</h1>
<h2 id="overview">Overview</h2>
<p>The Dual Mode Data Visualization (Metabolomics App) allows you to perform downstream analysis on single mode (either positive or negative mode) as well as dual mode (both positive and negative mode) targeted, semi-targeted (without retention time) and untargeted unlabeled metabolomics data along with insightful visualizations. The app provides a variety of normalization methods, scaling options and data visualization functionalities, thereby allowing an efficient analysis of the data to get actionable insights.</p>
<h2 id="scope-of-the-app">Scope of the App</h2>
<ul>
<li>The application supports data with a simple matrix having samples in the columns and metabolites in the rows.</li>
<li>It provides different normalization and scaling methods to perform on the data.</li>
<li>Performs quality checks for internal standards, metabolites, and samples.</li>
<li>Performs statistical analysis using limma and provides interactive visualizations.</li>
<li>Provides heatmap visualization along with different algorithms like hierarchical clustering, k-means, correlation etc.</li>
<li>Performs comparative analysis for the different cohort comparisons.</li>
</ul>
<p><img alt="Dual Mode Data Visualization (Metabolomics App) workflow schematic" src="../../img/DualMode/metab_app_workflow_schematics.png" /> <center><strong>Figure 1.</strong> Dual Mode Data Visualization </center></p>
<h1 id="getting-started">Getting Started</h1>
<h2 id="user-input">User Input</h2>
<p>To process single mode data, the following files are required:</p>
<ul>
<li>El-MAVEN Output File</li>
</ul>
<p><img alt="El-MAVEN raw intensity file obtained using a compound database" src="../../img/DualMode/metab_app_single_mode_elmaven.png" /> <center><strong>Figure 2.</strong> El-MAVEN raw intensity file obtained using a compound database</center></p>
<ul>
<li>El-MAVEN Internal Standards File (Optional)</li>
</ul>
<p><img alt="El-MAVEN raw intensity file of internal standards only" src="../../img/DualMode/metab_app_int_stds_elmaven.png" /> <center><strong>Figure 3.</strong> El-MAVEN raw intensity file of internal standards</center></p>
<ul>
<li>Cohort Mapping File</li>
</ul>
<p><img alt="Cohort file" src="../../img/DualMode/metab_app_single_mode_metadata.png" /> <center><strong>Figure 4.</strong> Cohort file</center></p>
<p>To process dual mode data, the following files are required:</p>
<ul>
<li>El-MAVEN Output Files from both positive and negative mode</li>
</ul>
<p><img alt="El-MAVEN raw intensity files for positive and negative modes" src="../../img/DualMode/metab_app_single_mode_elmaven.png" /></p>
<p><img alt="El-MAVEN raw intensity files for positive and negative modes" src="../../img/DualMode/metab_app_dual_mode_neg_elmaven.png" /> <center><strong>Figure 5.</strong> El-MAVEN raw intensity files for positive and negative modes</center></p>
<ul>
<li>El-MAVEN Internal Standards File from both, positive and negative modes (Optional)</li>
</ul>
<p><img alt="El-MAVEN raw intensity files for positive and negative modes of internal standards only" src="../../img/DualMode/metab_app_int_stds_elmaven.png" /></p>
<p><img alt="El-MAVEN raw intensity files for positive and negative modes of internal standards only" src="../../img/DualMode/metab_app_dual_mode_neg_stds_elmaven.png" /> <center><strong>Figure 6.</strong> El-MAVEN raw intensity files for positive and negative modes of internal standards only</center></p>
<ul>
<li>Cohort Mapping File</li>
</ul>
<p><img alt="Sample-Cohort metadata file" src="../../img/DualMode/metab_app_dual_mode_metadata.png" /> <center><strong>Figure 7.</strong> Sample-Cohort metadata file/center&gt;</center></p>
<p><strong>NOTE:</strong></p>
<ul>
<li>An already processed .gct file can also serve as input to the app.</li>
<li>The internal standard file is optional.</li>
</ul>
<h1 id="caveats">Caveats</h1>
<p>Pathways analysis only works when the data has KEGG Ids within the “compoundId” column.</p>
<h1 id="tutorial">Tutorial</h1>
<p>Select <em>Dual Mode Data Visualization (Metabolomics App)</em> from the dashboard under the <em>Metabolomics Data</em> Tab as shown in Figure 8. Create a <em>New Workspace</em> or choose from the existing one from the dop-down and provide the <em>Name of the Session</em> to be redirected to Dual Mode Data Visualisation (Metabolomics App)'s upload page.</p>
<p><img alt="Polly Dashboard and Workspace selection" src="../../img/DualMode/Dashboard1.png" /></p>
<p><img alt="Polly Dashboard and Workspace selection" src="../../img/DualMode/Selection1.png" /> <center><strong>Figure 8.</strong> Polly Dashboard and Workspace selection</center></p>
<h2 id="upload-files">Upload Files</h2>
<p>The Upload Files interface allows you to upload the input files required for processing through the app which includes the group summary matrix files from El-MAVEN and metadata file, or a .gct file.</p>
<p><img alt="Upload Files Tab" src="../../img/DualMode/metab_app_uploads_tab.png" /> <center><strong>Figure 9.</strong> Upload Files Tab</center></p>
<ul>
<li>
<p><em>Download all sample files:</em> This option would allow you to download the demo files which include the El-MAVEN output in group summary matrix format and cohort mapping file.</p>
</li>
<li>
<p><em>Upload GCT file:</em> Checking this box prompts the app that the input data is in the .gct format.</p>
</li>
</ul>
<p><img alt="Upload Files Tab: *Upload gct file* option is checked" src="../../img/DualMode/metab_app_uploads_gct_option.png" /> <center><strong>Figure 10.</strong> Upload Files Tab: <em>Upload gct file</em> option is checked</center></p>
<ul>
<li><em>Select mode:</em> This drop-down allows you to select the specific mode of the data in case of single mode, meaning whether it is:<ul>
<li>Positive, or</li>
<li>Negative</li>
</ul>
</li>
</ul>
<p><img alt="Upload Files Tab for single-mode data: default view" src="../../img/DualMode/metab_app_single_mode_uploads.png" /> <center><strong>Figure 11.</strong> Upload Files Tab for single-mode data: default view</center></p>
<p>To upload the output file of internal standards, click on <em>Upload internal standards file</em>.</p>
<p><img alt="Upload Files Tab for single-mode data: *Upload internal standards file* option is checked" src="../../img/DualMode/metab_app_single_mode_stds_option_uploads.png" /> <center><strong>Figure 12.</strong> Upload Files Tab for single-mode data: <em>Upload internal standards file</em> option is checked</center></p>
<ul>
<li>
<p><em>El-MAVEN positive/negative mode file:</em> This allows you to upload the positive/negative mode El-MAVEN output file (depending on the mode selected) in the .csv peak table format.</p>
</li>
<li>
<p>Internal Standards positive/negative mode file: This allows you to upload the positive/negative mode internal standards El-MAVEN output file (depending on the mode selected) in the .csv peak table format.</p>
</li>
<li>
<p><em>Metadata file:</em> This allows you to upload the cohort mapping file in the .csv format.</p>
</li>
<li>
<p><em>Data is dual mode:</em> Checking this box prompts the app that the input data is from dual-mode (positive and negative modes).</p>
</li>
</ul>
<p><img alt="Upload Files interface for dual-mode data" src="../../img/DualMode/metab_app_dual_mode_default_upload.png" /> <center><strong>Figure 13.</strong> Upload Files interface for dual-mode data</center></p>
<p>To upload the output file of internal standards, click on <em>Upload internal standards file</em>.</p>
<p><img alt="Upload Files Tab for dual-mode data: *Upload internal standards* *file* option is checked" src="../../img/DualMode/metab_app_dual_mode_stds_option_uploads.png" /> <center><strong>Figure 14.</strong> Upload Files Tab for dual-mode data: <em>Upload internal standards</em> <em>file</em> option is checked</center></p>
<ul>
<li>
<p><em>El-MAVEN negative mode file:</em> This allows you to upload the negative mode El-MAVEN output file in the .csv peak table format.</p>
</li>
<li>
<p><em>El-MAVEN positive mode file:</em> This allows you to upload the positive mode El-MAVEN output file in the .csv peak table format.</p>
</li>
<li>
<p><em>Internal Standards negative mode file (optional):</em> This allows you to upload the negative mode internal standards El-MAVEN output file in the .csv peak table format.</p>
</li>
<li>
<p><em>Internal Standards positive mode file (optional):</em> This allows you to upload the positive mode internal standards El-MAVEN output file in the .csv peak table format.</p>
</li>
<li>
<p><em>Metadata file:</em> This allows you to upload the cohort mapping file in the .csv format.</p>
</li>
</ul>
<p>Click on <em>Go</em> to proceed to the next step.</p>
<p><strong>Note:</strong></p>
<ul>
<li>The format of the metadata file for dual-mode should be in a specific format.</li>
</ul>
<p>To make common sample names across the different modes, click on <em>Replace samples to common sample names</em>.</p>
<p><img alt="Make common sample names: *Replace samples to common sample names* option is checked" src="../../img/DualMode/metab_app_common_samplenames.png" /> <center><strong>Figure 15.</strong> Make common sample names: <em>Replace samples to common sample names</em> option is checked</center></p>
<h2 id="pre-processing">Pre-processing</h2>
<p>The Pre-processing interface allows you to perform a multitude of functions on the data such as:</p>
<p><img alt="Preprocessing Tab" src="../../img/DualMode/metab_app_preprocessing_tab.png" /> <center><strong>Figure 16.</strong> Preprocessing Tab</center></p>
<ul>
<li><em>Select Internal Standards:</em> This allows you to select the internal standard(s) from within the El-MAVEN output file when a separate internal standards file is not provided as input.</li>
</ul>
<p><strong>Note:</strong></p>
<ul>
<li>In case, the internal standard(s) are not in the El-MAVEN output file but in the separate internal standards file, they will not show up in the drop down menu. To select the desired internal standards, select them in <em>Normalize by individual internal standards</em> option under <em>Normalize by Internal standards</em> in the <em>Perform Normalization &gt; Normalization</em>.</li>
</ul>
<p><img alt="Selecting internal standards from the data" src="../../img/DualMode/metab_app_int_stds_from_elmaven.png" /> <center><strong>Figure 17.</strong> Selecting internal standards from the data</center></p>
<ul>
<li><em>Drop Samples:</em> This allows you to drop/remove certain samples from further analysis which could be blank samples or any samples that didn’t have a good run during MS processing. Samples can be dropped by clicking on <em>Drop Samples</em> as shown in Figure 18 after selecting the sample(s) from the drop down menu.</li>
</ul>
<p><img alt="*Drop Samples* option" src="../../img/DualMode/metab_app_drop_samples_option.png" /> <center><strong>Figure 18.</strong> <em>Drop Samples</em> option</center></p>
<ul>
<li><em>Normalize by Internal standards</em> performs normalization using the internal standards.</li>
</ul>
<p><img alt="Normalize by internal standards options" src="../../img/DualMode/metab_app_norm_by_int_stds.png" /> <center><strong>Figure 19.</strong> Normalize by internal standards options</center></p>
<ul>
<li>
<p><em>Normalize by sum of internal standards</em> normalizes by the sum of the standards provided.</p>
</li>
<li>
<p><em>Normalize by average of internal standards</em> normalizes by the average of the standards provided.</p>
</li>
<li>
<p><em>Normalize by individual internal standards</em> normalizes by the internal standards selected previously.</p>
</li>
<li>
<p><em>Normalize by metabolites</em> normalizes by any particular metabolite selected.</p>
</li>
</ul>
<p><img alt="Normalize by metabolites option" src="../../img/DualMode/metab_app_normalise_by_metab_option.png" /> <center><strong>Figure 20.</strong> Normalize by metabolites option</center></p>
<ul>
<li>
<p><em>Normalize by sum of metabolites</em> normalizes by the sum of metabolites. Here, the user can select the metabolites from the dropdown option.</p>
</li>
<li>
<p><em>Normalize by metadata column</em> normalizes by any additional column specified in the metadata file. such as cell number etc.</p>
</li>
<li>
<p><em>Normalize by control</em> normalizes by control samples present in the data.</p>
</li>
</ul>
<p><img alt="Normalize by metadata options" src="../../img/DualMode/metab_app_normalize_by_metadata.png" /> <center><strong>Figure 21.</strong> Normalize by metadata options</center></p>
<ul>
<li>
<p><em>log2</em></p>
</li>
<li>
<p>y + log2(x) [where data is shifted by max value of data plus one]</p>
</li>
</ul>
<p><img alt="Scaling options" src="../../img/DualMode/metab_app_scaling_options.png" /> <center><strong>Figure 22.</strong> Scaling options</center></p>
<p><strong>Note:</strong></p>
<ul>
<li>If internal standard(s) have already been selected in the <em>Select Internal Standards</em> option, they would be present in the drop down.</li>
</ul>
<p>Clicking on <em>Run</em> will perform the normalization and scaling based on the parameters selected.</p>
<ul>
<li><em>Table:</em> This displays the data table and visualizations for both pre- and post- normalization.<ul>
<li><strong>Metadata:</strong> This displays the metadata uploaded. This data can be downloaded in the .csv format as shown in Figure 23.</li>
<li><strong>Metabolite Mapping data:</strong> This displays the metabolite data uploaded. This data can be downloaded in the .csv format as shown in Figure 24.</li>
<li><strong>Raw data:</strong> This displays the raw El-MAVEN data uploaded. This data can be downloaded in the .gct format as shown in Figure 25.</li>
<li><strong>Processed data:</strong> This displays the normalized El-MAVEN data based on the parameters selected. This data can be downloaded in .gct format as shown in Figure 26.</li>
<li><strong>Pre-Processing Results:</strong> This allows you to have a look at the sample distribution with the help of density plot and box-plot before normalization as shown in Figure 27.</li>
<li><strong>Post-Processing Results:</strong> This allows you to have a look at the sample distribution with the help of the density plot and box-plot after normalization. This provides you with the ability to check the effect of the normalization parameters on the data as shown in Figure 28.</li>
</ul>
</li>
</ul>
<p><img alt="Metadata table" src="../../img/DualMode/metab_app_metadata_table.png" /> <center><strong>Figure 23.</strong> Metadata table</center> </p>
<p><img alt="Metabolite Mapping data table" src="../../img/DualMode/metab_app_metabolite_mapping_data_table.png" /> <center><strong>Figure 24.</strong> Metabolite Mapping data table</center> </p>
<p><img alt="Raw data table" src="../../img/DualMode/metab_app_raw_data_table.png" /> <center><strong>Figure 25.</strong> Raw data table</center>  </p>
<p><img alt="Processed data table" src="../../img/DualMode/metab_app_normalized_data_table.png" /> <center><strong>Figure 26.</strong> Processed data table</center>  </p>
<p><img alt="Pre-Processing Results" src="../../img/DualMode/metab_app_pre_norm_results.png" /> <center><strong>Figure 27.</strong> Pre-Processing Results</center></p>
<p><img alt="Post-Processing Results" src="../../img/DualMode/metab_app_post_norm_results.png" /> <center><strong>Figure 28.</strong> Post-Processing Results</center></p>
<h2 id="quality-checks">Quality Checks</h2>
<p>This tab allows you to perform quality checks for the internal standards, metabolites and across samples with the help of interactive visualizations.</p>
<p><strong>Internal Standards</strong></p>
<p>It allows you to have a look at the quality of the internal standards used in the data with the help of the different visualizations for any individual as well as for all internal standards.</p>
<ul>
<li>Internal Standards (Individual): You can visualize the quality checks for any internal standard specifically. This allows you to select the internal standard by name, followed by another drop down to select by uniqueId of the feature. It’s also possible to specify the cohort order for the plots. For dual mode data, you can specify the internal standard of the particular mode from the <em>Select uniqueIds</em> drop down.</li>
</ul>
<p><img alt="Internal Standards (Individual) options" src="../../img/DualMode/metab_app_internal_standards_individual_page.png" /> <center><strong>Figure 29.</strong> Internal Standards (Individual) options</center></p>
<p><img alt="CV Distribution across cohorts" src="../../img/DualMode/metab_app_indi_stds_cv_distri_across_cohorts.png" /> <center><strong>Figure 30.</strong> CV Distribution across cohorts</center></p>
<p><img alt="CV Distribution across samples" src="../../img/DualMode/metab_app_indi_stds_cv_distri_samples.png" /> <center><strong>Figure 31.</strong> CV Distribution across samples</center></p>
<ul>
<li>
<p>Metabolites: It allows you to have a look at the quality of the metabolites present in the data with the help of the Coefficient of Variation plots</p>
<ul>
<li>Metabolites CoV Boxplot visualizes the Coefficient of Variation across different cohorts in the data in the form of the boxplot. It’s also possible to specify the cohort order for the plots as shown in Figure 32.</li>
<li>Metabolites CoV Barplot visualizes the Coefficient of Variation as a quality check for any specific metabolite. To use this, select the metabolite followed by the unique id of the feature using the drop downs shown in Figure 33. It’s also possible to specify the cohort order for the plots as shown in FIgure 33.</li>
</ul>
</li>
</ul>
<p><img alt="Metabolites CoV Boxplot option and CV Distribution across Cohorts boxplot" src="../../img/DualMode/metab_app_metabolites_cov_boxplot_options.png" /> <center><strong>Figure 32.</strong> Metabolites CoV Boxplot option and CV Distribution across Cohorts boxplot</center> </p>
<p><img alt="Metabolites CoV Barplot" src="../../img/DualMode/metab_app_metabolites_cov_barplot.png" /> <center><strong>Figure 33.</strong> Metabolites CoV Barplot</center></p>
<p><img alt="CV Distribution across cohorts for selected metabolite" src="../../img/DualMode/metab_app_metabolites_cv_barplot_cv_across_cohorts.png" /> <center><strong>Figure 34.</strong> CV Distribution across cohorts for selected metabolite</center></p>
<p><img alt="CV Distribution across samples for the selected metabolite" src="../../img/DualMode/metab_app_metabolites_cv_barplot_cv_across_samples.png" /> <center><strong>Figure 35.</strong> CV Distribution across samples for the selected metabolite</center></p>
<p><strong>PCA</strong></p>
<p>This allows you to understand the clustering pattern between biologically grouped and ungrouped samples.</p>
<ul>
<li>PCA (2D) provides PCA visualization in a two-dimensional manner by selecting the PC values for <em>x-</em> and <em>y-</em> axes. It’s also possible to specify the cohort order for the plots.</li>
</ul>
<p><img alt="Two-dimensional PCA plot" src="../../img/DualMode/metab_app_2d_pca.png" /> <center><strong>Figure 36.</strong> Two-dimensional PCA plot</center></p>
<ul>
<li>PCA (3D) provides PCA visualization in a three-dimensional manner by selecting the PC values for <em>x-</em>, <em>y-</em> and <em>z-</em> axes. It’s also possible to specify the cohort order for the plots.</li>
</ul>
<p><img alt="Three-dimensional PCA plot" src="../../img/DualMode/metab_app_3d_pca.png" /> <center><strong>Figure 37.</strong> Three-dimensional PCA plot</center></p>
<h2 id="statistical-analysis">Statistical Analysis</h2>
<p>This interface allows you to perform differential expression analysis with the aim to identify metabolites whose expression differs between any specified cohort conditions. The 'limma' R package is used to identify the differentially expressed metabolites. This method creates a log<sub>2</sub> fold change ratio between the two experimental conditions and an 'adjusted' <em>p</em>-value that rates the significance of the difference.</p>
<p><img alt="Statistical Analysis interface" src="../../img/DualMode/metab_app_stats_analysis_page.png" /> <center><strong>Figure 38.</strong> Statistical Analysis interface</center></p>
<p>The following parameters are available for selection:</p>
<ul>
<li>Select <em>Cohort A</em> and <em>Cohort B:</em> Default values are filled automatically for a selected cohort condition, which can be changed as per the cohorts of interest.</li>
<li>Select <em>p-val</em> or <em>adj. p-val</em>: Select either <em>p</em>-value or adj. <em>p</em>-value for significance.</li>
<li><em>p-val</em> or <em>adj. p-val</em> cutoff: By default, the value is 0.05 but can be changed if required.</li>
<li><em>log2FC:</em> Specify the cut-off for log<sub>2</sub> fold change with the help of the slider.</li>
</ul>
<p>Once the parameters are specified, click on the <em>Update</em> button to plot the volcano plot. Based on the parameters specified, a volcano plot is displayed. The volcano plot helps in visualizing metabolites that are significantly dysregulated between two cohorts.</p>
<p><img alt="Volcano plot" src="../../img/DualMode/metab_app_volcano_plot.png" /> <center><strong>Figure 39.</strong> Volcano plot</center></p>
<p><em>Filtered Metabolites Visualization</em> provides the visualization of cohort-based distribution of the metabolites that are significant based on the parameters specified.</p>
<p><img alt="Filtered Metabolites Visualization" src="../../img/DualMode/metab_app_stats_analysis_filtered_metabolites_visualization.png" /> <center><strong>Figure 40.</strong> Filtered Metabolites Visualization</center></p>
<p><em>Filtered Normalized Table</em> contains the normalized data of the metabolites that are significant based on the parameters specified.</p>
<p><img alt="Filtered Normalized Table" src="../../img/DualMode/metab_app_stats_analysis_filtered_normalised_table.png" /> <center><strong>Figure 41.</strong> Filtered Normalized Table</center></p>
<p><em>Filtered Differential Expression Table</em> contains only the metabolites that have significant <em>p</em>-values as specified.</p>
<p><img alt="Filtered Differential Expression Table" src="../../img/DualMode/metab_app_stats_analysis_filtered_diff_exp_table.png" /> <center><strong>Figure 42.</strong> Filtered Differential Expression Table</center></p>
<p><em>Differential Expression Table</em> contains all the differentially expressed metabolites without any filtering.</p>
<p><img alt="Differential Expression Table" src="../../img/DualMode/metab_app_stats_analysis_diff_exp_table.png" /> <center><strong>Figure 43.</strong> Differential Expression Table</center></p>
<p><em>Pathway Enrichment Analysis</em> performs the pathway enrichment analysis for the significant metabolites based on the parameters specified for the particular cohort comparison. Click on the <em>Perform Pathway Analysis</em> button.
As a result, you get Metabolite Set Enrichment Analysis and Pathway Topology Analysis plots that can be downloaded under the <em>Plot</em> panel. You can also obtain the tablular representation of the plots by selecting onto the <em>Table</em> panel.</p>
<p><img alt="Metabolite Set Enrichment Analysis Plot" src="../../img/DualMode/metab_app_stats_analysis_msea_plot.png" /> <center><strong>Figure 44.</strong> Metabolite Set Enrichment Analysis Plot</center></p>
<p><img alt="Pathway Topology Analysis Plot" src="../../img/DualMode/metab_app_stats_analysis_pathway_topology_plot.png" /> <center><strong>Figure 45.</strong> Pathway Topology Analysis Plot</center></p>
<p><img alt="Metabolite Set Enrichment Analysis Table" src="../../img/DualMode/metab_app_stats_analysis_msea_table.png" /> <center><strong>Figure 46.</strong> Metabolite Set Enrichment Analysis Table</center></p>
<p><img alt="Pathway Topology Analysis Table" src="../../img/DualMode/metab_app_stats_analysis_topology_table.png" /> <center><strong>Figure 47.</strong> Pathway Topology Analysis Table</center></p>
<p><em>Pathway View</em> plots the pathway view of the metabolites that show up in the Metabolite Set Enrichment Analysis. It maps and renders the metabolite hits on relevant pathway graphs. This enables you to visualize the significant metabolites on pathway graphs of the respective metabolites they belong to. You can select your metabolite of interest from the drop-down and click on <em>Plot</em>. This will plot the pathway view of the metabolism selected. You can also download the plot as a .png file by clicking onto the <em>Download Pathview Plot</em> button.</p>
<p><img alt="Pathway View Plot" src="../../img/DualMode/metab_app_stats_analysis_pathview_plot.png" /> <center><strong>Figure 48.</strong> Pathway View Plot</center></p>
<h2 id="visualization">Visualization</h2>
<p>This interface allows you to visualize the cohort-based distribution of a specific metabolite or a group of metabolites on the basis on its normalized intensity values.</p>
<p><img alt="Visualization tab options" src="../../img/DualMode/metab_app_viz_tab_options.png" /> <center><strong>Figure 49.</strong> Visualization tab options</center></p>
<ul>
<li><em>Enter metabolite:</em> Select the metabolite(s) of interest from the drop down option.</li>
<li><em>Select uniqueIds:</em> You can specifically select the metabolic feature of interest for the metabolite from the drop down option.</li>
<li><em>Select order of cohort:</em> You can also specify the particular order of the cohort to visualize the bar plot.</li>
</ul>
<p>Once the parameters are selected, click on <em>Load Plots</em> to plot the bar plot for the metabolite.</p>
<p><img alt="Cohort-wise bar plot with the normalized intensity of selected metabolite" src="../../img/DualMode/metab_app_barplot_viz.png" /> <center><strong>Figure 50.</strong> Cohort-wise bar plot with the normalized intensity of selected metabolite</center></p>
<h2 id="intomix-input">IntOmix Input</h2>
<p>This tab allows you to generate the input for <a href="https://docs.elucidata.io/Apps/Multi-omic%20Data/IntOmix.html">IntOmix</a> where you can visualize the significantly altered metabolic network modules between any two experimental conditions.</p>
<p><img alt="IntOmix Input Tab options" src="../../img/DualMode/metab_app_intomix_input_tab.png" /> <center><strong>Figure 51.</strong> IntOmix Input Tab options</center></p>
<p>Specify two or more cohorts from the <em>Select cohorts</em> drop down option for which you want to generate the IntOmix input. Once the required cohorts are selected, click on <em>Generate</em> to generate the IntOmix input.</p>
<p><img alt="IntOmix Table for the cohort conditions specified" src="../../img/DualMode/metab_app_intomix_table.png" /> <center><strong>Figure 52.</strong> IntOmix Table for the cohort conditions specified</center></p>
<p><strong>NOTE:</strong></p>
<ul>
<li>At least two cohorts are required to create the input file.</li>
</ul>
<h2 id="heatmap">Heatmap</h2>
<p>This tab allows you to produce a heatmap of the processed data, so that you can observe the level of expression in a visual form. Click on <em>Load Heatmap</em> button to generate the heatmap.</p>
<p><img alt="Heatmap of the processed data" src="../../img/DualMode/metab_app_heatmap.png" /> <center><strong>Figure 53.</strong> Heatmap</center></p>
<h2 id="comparative-analysis">Comparative Analysis</h2>
<p>This tab allows you to perform comparative analysis between a set of cohorts in your data. As a result of which you can visualize the UpSet plot of the unique and overlapping metabolites for the selected cohort comparisons. Further, you can also perform pathway analysis on the metabolites for the set intersections of interest.</p>
<ul>
<li><em>Comparison Parameters</em> tab allows you to select the cohorts of interest for which you would want to get the set intersections. You can select the cohorts from the <em>Select cohorts</em> drop-down and click on <em>Run</em> button. Further, you can also specify the <em>p</em>-value cut-off and log<sub>2</sub>FC threshold.</li>
</ul>
<p><img alt="Comparison Parameters Tab options" src="../../img/DualMode/metab_app_comparative_analysis_uploads_tab.png" /> <center><strong>Figure 54.</strong> Comparison Parameters Tab options</center></p>
<p>You will get a table as a result of the parameters specified which will have the significant metabolites for the different cohort comparisons along with their corresponding <em>p</em>-values and log<sub>2</sub>FC values. You can also download this table as a .CSV file.</p>
<p><img alt="Comparison Table" src="../../img/DualMode/metab_app_comparative_analysis_comparison_table.png" /> <center><strong>Figure 55.</strong> Comparison Table</center></p>
<ul>
<li><em>UpSet Plot</em> tab allows you to visualize the set intersections for the cohort comparisons selected where every comparison consists of the significant metabolites associated with the same. You can select the cohort comparisons of interest from the <em>Select Cohort Comparison</em> drop-down which represents all the possible comparisons for the cohorts specified in the previous tab. Click on <em>Plot</em> to get the UpSet plot for the specified comparisons.</li>
</ul>
<p><img alt="UpSet Plot options" src="../../img/DualMode/metab_app_comparative_analysis_upset_plot_options.png" /> <center><strong>Figure 56.</strong> UpSet Plot options</center></p>
<p><img alt="UpSet Plot" src="../../img/DualMode/metab_app_comparative_analysis_upset_plot.png" /> <center><strong>Figure 57.</strong> UpSet Plot</center></p>
<p>Along with the plot, you can also get all the constituent metabolites for the respective comparisons in a tabular format that can be downloaded as a .CSV file.</p>
<p><img alt="UpSet Plot table" src="../../img/DualMode/metab_app_comparative_analysis_upset_table.png" /> <center><strong>Figure 58.</strong> UpSet Plot Table</center></p>
<ul>
<li><em>Pathway Enrichment Analysis</em> tab allows you to perform the pathway enrichment analysis for the significant metabolites that show up based on the parameters specified in the <em>Comparison Parameters</em> tab for the particular set of cohort comparison. Click on the <em>Perform Pathway Analysis</em> button. As a result, you get Metabolite Set Enrichment Analysis and Pathway Topology Analysis plots that can be downloaded under the <em>Plot</em> panel. You can also obtain the tablular representation of the plots by selecting onto the <em>Table</em> panel.</li>
</ul>
<p><img alt="Metabolite Set Enrichment Analysis Plot" src="../../img/DualMode/metab_app_comparative_analysis_msea_plot.png" /> <center><strong>Figure 59.</strong> Metabolite Set Enrichment Analysis Plot</center></p>
<p><img alt="Pathway Topology Analysis Plot" src="../../img/DualMode/metab_app_comparative_analysis_pathway_topology_plot.png" /> <center><strong>Figure 60.</strong> Pathway Topology Analysis Plot</center></p>
<p><img alt="Metabolite Set Enrichment Analysis Table" src="../../img/DualMode/metab_app_comparative_analysis_msea_table.png" /> <center><strong>Figure 61.</strong> Metabolite Set Enrichment Analysis Table</center></p>
<p><img alt="Pathway Topology Analysis Table" src="../../img/DualMode/metab_app_comparative_analysis_topology_table.png" /> <center><strong>Figure 62.</strong> Pathway Topology Analysis Table</center></p>
<p><em>Pathway View</em> plots the pathway view of the metabolites that show up in the Metabolite Set Enrichment Analysis. It maps and renders the metabolite hits on relevant pathway graphs. This enables you to visualize the significant metabolites on pathway graphs of the respective metabolisms they belong to. You can select your metabolite of interest from the drop-down and click on <em>Plot</em>. This will plot the pathway view of the metabolite selected. You can also download the plot as a .png file by clicking onto the <em>Download Pathview Plot</em> button.</p>
<p><img alt="Pathway View options" src="../../img/DualMode/metab_app_comparative_analysis_pathway_view_options.png" /> <center><strong>Figure 63.</strong> Pathway View options</center></p>
<p><img alt="Pathway View Plot" src="../../img/DualMode/metab_app_comparative_analysis_pathway_view_plot.png" /> <center><strong>Figure 64.</strong> Pathway View Plot</center></p>
<h2 id="data-for-metaboanalyst">Data for MetaboAnalyst</h2>
<p>This tab allows you to generate the input for MetaboAnalyst that enables statistical, functional and integrative analysis of metabolomics data by providing a variety of modules for different functionalities.</p>
<p><img alt="Data for MetaboAnalyst Tab options" src="../../img/DualMode/metab_app_metaboanalyst_page.png" /> <center><strong>Figure 65.</strong> <em>Data for MetaboAnalyst</em> Tab options</center></p>
<p>Specify the cohort comparison of interest by sleecting the cohorts from the <em>Select cohort A</em> and <em>Select cohort B</em> drop downs. Click on <em>Download data for metaboanalyst</em> to download the .csv file. Clicking on <em>Go to metaboanalyst</em> will redirect you to MetaboAnalyst’s homepage.</p>
<p><strong>NOTE:</strong></p>
<ul>
<li>Normalized intensity data (normalization method chosen in pre-processing tab) is downloaded.</li>
</ul>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="QuantFit.html" class="btn btn-neutral float-right" title="QuantFit">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="Untargeted Pipeline.html" class="btn btn-neutral" title="Untargeted Pipeline"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="Untargeted Pipeline.html" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="QuantFit.html" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '../..';</script>
    <script src="../../js/theme.js" defer></script>
      <script src="../../search/main.js" defer></script>

</body>
</html>
